<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application to single-cell RNA-seq data • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Application to single-cell RNA-seq data">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_highD_default.html">Multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_highD_advanced.html">Advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_spatial_transcriptomics.html">Spatial transcriptomics</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_moca_100k.html">MOCA 100k cells</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_example_tsne2D_default.html">2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a07_example_tsne2D_advanced.html">Advanced mode on 2D t-SNE coordinates</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Application to single-cell RNA-seq data</h1>
                        <h4 data-toc-skip class="author">Alexis
Vandenbon</h4>
            
            <h4 data-toc-skip class="date">2022-11-04</h4>
      
      
      <div class="hidden name"><code>a02_example_scRNAseq.Rmd</code></div>

    </div>

    
    
<p>In this article, we show the application of
<code>singleCellHaystack</code> to predict differentially expressed
genes (DEGs) in a single-cell RNA-seq dataset. The input data is 1) the
gene expression data and 2) the coordinates of each cell in an input
space.</p>
<p>In this example, the gene expression data is an ordinary matrix
(genes versus cells), but a Seurat object (for example) could also be
used. The input space in this example will be the first principal
components of the cells. <code>singleCellHaystack</code> predicts DEGs
by looking at the “expression distribution” of a gene (i.e. the
distribution in the input space of cells that express a gene).</p>
<div class="section level2">
<h2 id="preparing-input-data">Preparing input data<a class="anchor" aria-label="anchor" href="#preparing-input-data"></a>
</h2>
<p>The data used in these examples can be found <a href="https://figshare.com/articles/dataset/Example_datasets_for_singleCellHaystack_GitHub_vignettes_2022_/21499071" class="external-link">here</a>.
We recommend downloading the .rda file and loading it using the
<code><a href="https://rdrr.io/r/base/load.html" class="external-link">load()</a></code> function. You can also download the individual data
files separately.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://figshare.com/ndownloader/files/38094522"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/ls.html" class="external-link">ls</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "dat.expression" "dat.pca"        "dat.tsne"</span></span></code></pre></div>
<p>This data should include the following objects:</p>
<ul>
<li>
<code>dat.expression</code>: a matrix object with the expression of
genes (rows) in each cell (columns).</li>
<li>
<code>dat.pca</code>: the output of PCA. This data.frame contains
the first 20 pricipal components (PCs).</li>
<li>
<code>dat.tsne</code>: a data.frame with t-SNE coordinates (2D)
based on the first 20 PCs.</li>
</ul>
<p>Let’s have a look at this example dataset:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># this data contains 10,537 genes and 1,429 cells</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 10537  1429</span></span>
<span></span>
<span><span class="co"># visualizing the cells in a t-SNE plot:</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tSNE_1</span>, y <span class="op">=</span> <span class="va">tSNE_2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a02_example_scRNAseq_files/figure-html/unnamed-chunk-2-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># the t-SNE coordinates are based on the 20 PCs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dat.pca</span><span class="op">)</span> <span class="co"># the PCA data contains 1,429 cells and 20 PCs</span></span>
<span><span class="co">#&gt; [1] 1429   20</span></span></code></pre></div>
<p>There are several groups of cells, although the borders between them
are not clear and several might consist of additional subclusters. We
can use <code>singleCellHaystack</code> to predict DEGs without relying
on grouping of the cells into arbitrary clusters.</p>
</div>
<div class="section level2">
<h2 id="running-haystack-on-multi-dimensional-coordinates">Running <code>haystack</code> on multi-dimensional coordinates<a class="anchor" aria-label="anchor" href="#running-haystack-on-multi-dimensional-coordinates"></a>
</h2>
<p>First, load the package.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span></span></code></pre></div>
<p>Next, run <code>haystack</code> on the first 20 principal components.
We also give the expression data as input. This example dataset is
relatively small, containing 1,429 cells, so running ‘haystack’ should
take just 1 to 3 minutes to finish. We also set a random seed to ensure
replicability.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">res.pc20</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/haystack.html">haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.pca</span>, expression <span class="op">=</span> <span class="va">dat.expression</span><span class="op">)</span></span>
<span><span class="co">#&gt; ### calling haystack_continuous_highD()...</span></span>
<span><span class="co">#&gt; ### Using package sparseMatrixStats to speed up statistics in sparse matrices.</span></span>
<span><span class="co">#&gt; ### Calculating row-wise mean and SD...</span></span>
<span><span class="co">#&gt; ### Filtered 0 genes with zero variance...</span></span>
<span><span class="co">#&gt; ### Using 100 randomizations...</span></span>
<span><span class="co">#&gt; ### Using 100 genes to randomize...</span></span>
<span><span class="co">#&gt; ### scaling input data...</span></span>
<span><span class="co">#&gt; ### deciding grid points...</span></span>
<span><span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span></span>
<span><span class="co">#&gt; ### performing randomizations...</span></span>
<span><span class="co">#&gt; ### estimating p-values...</span></span>
<span><span class="co">#&gt; ### picking model for mean D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.035</span></span>
<span><span class="co">#&gt; ### best df    : 5</span></span>
<span><span class="co">#&gt; ### picking model for stdev D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.032</span></span>
<span><span class="co">#&gt; ### best df    : 10</span></span>
<span><span class="co">#&gt; ### returning result...</span></span></code></pre></div>
<p>Let’s have a look at the most significant DEGs. The gene with the
strongest differential expression is Cd53. We can plot the expression
and detection of this gene using the <code>plot_gene_haystack</code>
function. From the t-SNE plots we can see that Cd53 is expressed in
cells at the bottom left. In contrast, the expression of Top2a is
limited to a different subset of cells, on the right.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res.pc20</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     D_KL log.p.vals log.p.adj</span></span>
<span><span class="co">#&gt; Cd53          0.08172190  -64.78888 -60.76616</span></span>
<span><span class="co">#&gt; Shisa5        0.03972582  -62.84831 -58.82559</span></span>
<span><span class="co">#&gt; H2-K1         0.04427821  -60.10620 -56.08348</span></span>
<span><span class="co">#&gt; Gimap4        0.11147736  -52.47112 -48.44840</span></span>
<span><span class="co">#&gt; Ms4a4b        0.09291971  -50.40240 -46.37968</span></span>
<span><span class="co">#&gt; Top2a         0.03324626  -48.09147 -44.06875</span></span>
<span><span class="co">#&gt; Evl           0.05532307  -47.88664 -43.86392</span></span>
<span><span class="co">#&gt; 2810417H13Rik 0.02536247  -47.83085 -43.80813</span></span>
<span><span class="co">#&gt; Ifi27l2a      0.05388583  -47.82993 -43.80721</span></span>
<span><span class="co">#&gt; Ccnd2         0.10407929  -47.80352 -43.78081</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># prepare a data.frame for plotting</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">d</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                tSNE_1      tSNE_2    Mrpl15    Lypla1</span></span>
<span><span class="co">#&gt; 10X_P7_11_AAACCTGAGACAGGCT -10.461888 -11.4182096 2.0819393 0.0000000</span></span>
<span><span class="co">#&gt; 10X_P7_11_AAACCTGAGAGTCTGG   7.392279  -5.2816862 0.5902685 0.3380684</span></span>
<span><span class="co">#&gt; 10X_P7_11_AAACCTGCAGGAATGC -19.930710   0.3160883 0.0000000 0.0000000</span></span>
<span><span class="co">#&gt; 10X_P7_11_AAACCTGGTACCGTAT -16.321436  15.4128833 1.5215021 1.5215021</span></span>
<span></span>
<span><span class="co"># plotting gene Cd53</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">d</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">tSNE_1</span>, <span class="va">tSNE_2</span>, color<span class="op">=</span><span class="va">Cd53</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_distiller</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Spectral"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a02_example_scRNAseq_files/figure-html/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># plotting gene Top2a</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">d</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">tSNE_1</span>, <span class="va">tSNE_2</span>, color<span class="op">=</span><span class="va">Top2a</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_distiller</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Spectral"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a02_example_scRNAseq_files/figure-html/unnamed-chunk-6-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="clustering-and-visualization">Clustering and visualization<a class="anchor" aria-label="anchor" href="#clustering-and-visualization"></a>
</h2>
<p>Next, let’s take the top 1000 DEGs, and cluster them by their
expression pattern in the input space (first 20 PCs). Here we use
<code>hclust_haystack</code>, which uses hierarchical clustering.
Alternatively, we could use <code>kmeans_haystack</code> for k-means
clustering.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the 1000 most significant DEGs, and cluster them by their distribution pattern in the 2D plot</span></span>
<span><span class="va">sorted.table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res.pc20</span>, n <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">gene.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">sorted.table</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># cluster the genes by their expression pattern in the input space using hierarchical clustering</span></span>
<span><span class="va">res.hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/hclust_haystack.html">hclust_haystack</a></span><span class="op">(</span><span class="va">dat.pca</span>, <span class="va">dat.expression</span><span class="op">[</span><span class="va">gene.subset</span>, <span class="op">]</span>, grid.coordinates<span class="op">=</span><span class="va">res.pc20</span><span class="op">$</span><span class="va">info</span><span class="op">$</span><span class="va">grid.coordinates</span><span class="op">)</span></span>
<span><span class="co">#&gt; ### collecting density data...</span></span></code></pre></div>
<p><code>hclust_haystack</code> returns as result a <code>hclust</code>
tree, which we can cut into clusters using the <code>cutree</code>
function. Here, we arbitrarily set the number of clusters to 5.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.hc.clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html" class="external-link">cutree</a></span><span class="op">(</span><span class="va">res.hc</span>, k<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">res.hc.clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; res.hc.clusters</span></span>
<span><span class="co">#&gt;   1   2   3   4   5 </span></span>
<span><span class="co">#&gt;  88  90 605 188  29</span></span></code></pre></div>
<p>While some clusters contain hundreds of genes, others are quite
small.</p>
<p>Next, let’s calculate the average expression of the genes in each
cluster.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">cluster</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">res.hc.clusters</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">d</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cluster_"</span>, <span class="va">cluster</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res.hc.clusters</span> <span class="op">==</span> <span class="va">cluster</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Let’s run through the 5 clusters and plot their averaged detection
pattern using <code>plot_gene_set_haystack</code>, which is similar to
<code>plot_gene_haystack</code> but uses a set of genes as input instead
of just 1 gene.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster_1"</span>, <span class="st">"cluster_2"</span>, <span class="st">"cluster_3"</span>, <span class="st">"cluster_4"</span>, <span class="st">"cluster_5"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="va">d</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">tSNE_1</span>, <span class="va">tSNE_2</span>, color<span class="op">=</span><span class="va">.data</span><span class="op">[[</span><span class="va">cluster</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_distiller</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Spectral"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a02_example_scRNAseq_files/figure-html/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;">
As we can see from these plots, each cluster has a distinct pattern.
Some clusters have high expression over a broad range of cells, while
cluster 5 is limited to a smaller subset.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res.hc.clusters</span><span class="op">[</span><span class="st">"Cd53"</span><span class="op">]</span></span>
<span><span class="co">#&gt; Cd53 </span></span>
<span><span class="co">#&gt;    1</span></span></code></pre></div>
<p>The most significant DEG, Cd53, was clustered into cluster 1.
Comparing its expression pattern (see above) with that of each cluster,
we can indeed see that it fits most closely with that cluster.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
