<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application of 'advanced' mode on multi-dimensional coordinates • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Application of 'advanced' mode on multi-dimensional coordinates">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_highD_default.html">Multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_highD_advanced.html">Advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_spatial_transcriptomics.html">Spatial transcriptomics</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_moca_100k.html">MOCA 100k cells</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_example_tsne2D_default.html">2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a07_example_tsne2D_advanced.html">Advanced mode on 2D t-SNE coordinates</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a03_example_highD_advanced_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Application of ‘advanced’ mode on multi-dimensional coordinates</h1>
                        <h4 class="author">Alexis Vandenbon</h4>
            
            <h4 class="date">2021-03-29</h4>
      
      
      <div class="hidden name"><code>a03_example_highD_advanced.Rmd</code></div>

    </div>

    
    
<div id="tldr" class="section level1">
<h1 class="hasAnchor">
<a href="#tldr" class="anchor"></a>tl;dr</h1>
<p>You can indicate the general expression levels of cells using input parameter <code>use.advanced.sampling</code>. If you do that,<code>singleCellHaystack</code> takes the general expression levels into account and finds genes (DEGs) that are different from that general expression pattern.</p>
</div>
<div id="preparing-input-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preparing-input-data" class="anchor"></a>Preparing input data</h1>
<p>The data used in these examples can be found <a href="https://figshare.com/articles/dataset/Example_dataset_for_singleCellHaystack_GitHub_vignettes/12923489">here</a>. We recommend downloading the .rda file and loading it using the <code><a href="https://rdrr.io/r/base/load.html">load()</a></code> function. You can also download the individual data files separately.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span><span class="op">(</span><span class="st">"https://ndownloader.figshare.com/files/24593213"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] "dat.expression" "dat.pca"        "dat.tsne"       "dat.umap"</span></code></pre></div>
<p>This data should include the following objects:</p>
<ul>
<li>
<code>dat.expression</code>: a matrix object with the expression of genes (rows) in each cell (columns).</li>
<li>
<code>dat.pca</code>: the output of PCA. This data.frame contains the first 50 pricipal components (PCs).</li>
<li>
<code>dat.tsne</code>: a data.frame with t-SNE coordinates (2D) based on the first 50 PCs.</li>
<li>
<code>dat.umap</code>: a data.frame with UMAP coordinates (2D) based on the first 50 PCs.</li>
</ul>
<p>Let’s have a look at this example dataset:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># this data contains 12,030 genes and 1981 cells</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">)</span>
<span class="co">#&gt; [1] 12030  1981</span>

<span class="co"># visualizing the cells in a t-SNE plot:</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">V1</span>, y <span class="op">=</span> <span class="va">V2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"t-SNE1"</span>, y <span class="op">=</span> <span class="st">"t-SNE2"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example1-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># the t-SNE coordinates are based on the 50 PCs</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">dat.pca</span><span class="op">)</span> <span class="co"># the PCA data contains 1981 cells and 50 PCs</span>
<span class="co">#&gt; [1] 1981   50</span></code></pre></div>
<p>There are several groups of cells, although the borders between them are not clear and several might consist of additional subclusters. We can use <code>singleCellHaystack</code> to predict DEGs without relying on grouping of the cells into arbitrary clusters.</p>
<p><code>singleCellHaystack</code> has two required input parameters: 1) the coordinates of cells in the input space, and 2) a table showing which genes are detected in which cells. The definition of detection is left to the user. In this example we will use the median normalized read count of each gene as threshold for defining detection. Alternatively, we could also define genes with counts &gt; 0 as being detected, for example.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">median.per.gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">dat.expression</span>,<span class="fl">1</span>,<span class="va">median</span><span class="op">)</span> <span class="co"># get the median read count per gene</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">median.per.gene</span><span class="op">)</span> <span class="co"># for many genes the median read count is 0</span>
<span class="co">#&gt;  Mrpl15  Lypla1  Rb1cc1    Sgk3   Snhg6   Tram1 </span>
<span class="co">#&gt;   0.000   0.000   0.000   0.000   0.000 136.243</span>
<span class="va">dat.detection</span> <span class="op">&lt;-</span> <span class="va">dat.expression</span> <span class="op">&gt;</span> <span class="va">median.per.gene</span> <span class="co"># use the medians as threshold for defining detection</span>
<span class="va">dat.detection</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span> <span class="co"># TRUE means detected, FALSE means not detected</span>
<span class="co">#&gt;        AAACCTGCAGTAACGG.1 AAACGGGAGAAGAAGC.1 AAACGGGAGACCGGAT.1</span>
<span class="co">#&gt; Mrpl15               TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Lypla1               TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Rb1cc1               TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Sgk3                 TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Snhg6                TRUE              FALSE              FALSE</span>
<span class="co">#&gt;        AAACGGGAGCGCTCCA.1 AAACGGGAGGACGAAA.1</span>
<span class="co">#&gt; Mrpl15              FALSE               TRUE</span>
<span class="co">#&gt; Lypla1              FALSE               TRUE</span>
<span class="co">#&gt; Rb1cc1              FALSE              FALSE</span>
<span class="co">#&gt; Sgk3                FALSE              FALSE</span>
<span class="co">#&gt; Snhg6                TRUE              FALSE</span></code></pre></div>
<p>Now, that we have defined gene detection, let’s have a look at the t-SNE plot again, colouring cells by the number of genes detected in each cell:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">general.detection</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">dat.detection</span>, <span class="fl">2</span>, <span class="va">sum</span><span class="op">)</span>
<span class="fu">ggplot</span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">V1</span>, y <span class="op">=</span> <span class="va">V2</span>, colour <span class="op">=</span> <span class="va">general.detection</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"t-SNE1"</span>, y <span class="op">=</span> <span class="st">"t-SNE2"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_color_gradient</span><span class="op">(</span>low<span class="op">=</span><span class="st">"dodgerblue"</span>, high<span class="op">=</span><span class="st">"maroon2"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Det. genes"</span><span class="op">)</span></code></pre></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example3-1.png" width="576" style="display: block; margin: auto;"></p>
<p>As can be seen in the last plot, cells in the bottom right have in general fewer detected genes. If a gene is detected predominantly in these cells and not in others, this tendency goes against the general trend in expression. Such a trend could be of particular interest.</p>
<p>We can use <code>singleCellHaystack</code> to predict DEGs without relying on grouping of the cells into arbitrary clusters, taking also into account general gene expression trends, using the option <code>use.advanced.sampling</code>.</p>
</div>
<div id="running-haystack-with-use-advanced-sampling" class="section level1">
<h1 class="hasAnchor">
<a href="#running-haystack-with-use-advanced-sampling" class="anchor"></a>Running <code>haystack</code> with <code>use.advanced.sampling</code>
</h1>
<p>First, load the package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span></code></pre></div>
<p>Next, run <code>haystack</code> on the first 50 principal components. Since the space is 50-dimensional, we set ‘method’ to ‘highD’. We also give the detection values as input to ‘detection’. This example dataset is relatively small, containing 1,981 cells, so running ‘haystack’ should take just 1 to 3 minutes to finish. We also give the detection values as input to ‘detection’. <code>use.advanced.sampling</code> is set to the vector of detected genes per cell, <code>general.detection</code>. By doing this, <code>haystack</code> will calculate cell distributions and perform randomizations taking into account the general detection levels. We also set a random seed to ensure replicability.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">res.pc50.adv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/haystack.html">haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.pca</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>, detection <span class="op">=</span> <span class="va">dat.detection</span>, method <span class="op">=</span> <span class="st">"highD"</span>, use.advanced.sampling <span class="op">=</span> <span class="va">general.detection</span><span class="op">)</span>
<span class="co">#&gt; ### calling haystack_highD()...</span>
<span class="co">#&gt; ### Using package wrswoR to speed up random sampling</span>
<span class="co">#&gt; ### scaling input data...</span>
<span class="co">#&gt; ### deciding grid points...</span>
<span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span>
<span class="co">#&gt; ### performing randomizations...</span>
<span class="co">#&gt; ### estimating p-values...</span>
<span class="co">#&gt; ### returning result...</span></code></pre></div>
<p>Let’s have a look at the most significant DEGs. The gene with the strongest differential expression is Tspo2. We can plot the expression and detection of this gene using the <code>plot_gene_haystack</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res.pc50.adv</span>, n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt;                     D_KL log.p.vals log.p.adj T.counts</span>
<span class="co">#&gt; Tspo2         0.01876159  -3943.000 -3938.920      299</span>
<span class="co">#&gt; Fam46c        0.01742252  -3898.379 -3894.298      283</span>
<span class="co">#&gt; 2410076I21Rik 0.01960576  -3742.188 -3738.108      245</span>
<span class="co">#&gt; Cd36          0.01661617  -3406.159 -3402.078      251</span>
<span class="co">#&gt; Cdr2          0.01710636  -3173.139 -3169.059      241</span>
<span class="co"># plotting detection of this gene</span>
<span class="fu"><a href="../../reference/plot_gene_haystack.html">plot_gene_haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.tsne</span>, gene <span class="op">=</span> <span class="st">"Tspo2"</span>, expression <span class="op">=</span> <span class="va">dat.detection</span><span class="op">)</span></code></pre></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example6-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plotting log expression of this gene</span>
<span class="fu"><a href="../../reference/plot_gene_haystack.html">plot_gene_haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.tsne</span>, gene <span class="op">=</span> <span class="st">"Tspo2"</span>, expression <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example6-2.png" width="576" style="display: block; margin: auto;"></p>
<p>From the plots we can see that Tspo2 is detected in the lower right groups of cells. As seen above, these cells have in general fewer detected genes, so the pattern of Tspo2 is especially surprizing: it is deteced in cells that express in general few genes.</p>
<p>This explains why the advanced mode of <code>singleCellHaystack</code> judges Tspo2 to be the most significant DEG.</p>
<p>The expression distribution of Tspo2 is similar to that of Trim10, the top DEG predicted in the default mode (see <a href="a02_example_highD_default.html">here</a>). Indeed Trim10 ranked among the top DEGs here too (it ranks 7th). On the other hand, other top DEGs of the “default” mode are detected in cells that have in general many detected genes. Hence, they are not among the top DEGs in the “advanced” mode.</p>
</div>
<div id="clustering-and-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering-and-visualization" class="anchor"></a>Clustering and visualization</h1>
<p>Next, let’s take the top 1000 DEGs, and cluster them by their expression pattern in the input space (first 50 PCs). Here we use <code>hclust_haystack</code>, which uses hierarchical clustering. Alternatively, we could use <code>kmeans_haystack</code> for k-means clustering.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get the top 1000 DEGs in the result</span>
<span class="va">res.top</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res.pc50.adv</span>, n <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>
<span class="co"># cluster DEGs by their expression pattern in the 2D plot</span>
<span class="va">genes.top</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">res.top</span><span class="op">)</span>
<span class="va">res.hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/hclust_haystack.html">hclust_haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.tsne</span>, genes <span class="op">=</span> <span class="va">genes.top</span>, detection <span class="op">=</span> <span class="va">dat.detection</span><span class="op">)</span></code></pre></div>
<p><code>hclust_haystack</code> returns as result a <code>hclust</code> tree, which we can cut into clusters using the <code>cutree</code> function. Here, we arbitrarily set the number of clusters to 5.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res.hc.clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span><span class="op">(</span><span class="va">res.hc</span>, k<span class="op">=</span><span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">res.hc.clusters</span><span class="op">)</span>
<span class="co">#&gt; res.hc.clusters</span>
<span class="co">#&gt;   1   2   3   4   5 </span>
<span class="co">#&gt; 159  85 225 367 164</span></code></pre></div>
<p>Let’s run through the 5 clusters and plot their averaged detection pattern using <code>plot_gene_set_haystack</code>, which is similar to <code>plot_gene_haystack</code> but uses a set of genes as input instead of just 1 gene.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="kw">function</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">gene.set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">res.hc.clusters</span><span class="op">)</span><span class="op">[</span><span class="va">res.hc.clusters</span><span class="op">==</span><span class="va">cluster</span><span class="op">]</span>
  <span class="va">plot.title</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Cluster "</span>, <span class="va">cluster</span><span class="op">)</span>
  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/plot_gene_set_haystack.html">plot_gene_set_haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.tsne</span>, genes <span class="op">=</span> <span class="va">gene.set</span>, detection <span class="op">=</span> <span class="va">dat.detection</span><span class="op">)</span>
  <span class="va">p</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="va">plot.title</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu">plot_grid</span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">pl</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example9-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res.hc.clusters</span><span class="op">[</span><span class="st">"Tspo2"</span><span class="op">]</span> <span class="co"># the top DEG is in cluster 5</span>
<span class="co">#&gt; Tspo2 </span>
<span class="co">#&gt;     5</span></code></pre></div>
<p>The most significant DEG, Tspo2, was clustered into cluster 5. Comparing its expression pattern (see above) with that of each cluster, we can indeed see that it fits most closely with that of cluster 5. Cluster 5 genes are on average detected almost exclusively in the bottom right group of cells.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
