<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of spatial transcriptomics with singleCellHaystack â€¢ singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Analysis of spatial transcriptomics with singleCellHaystack">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_highD_default.html">Multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_highD_advanced.html">Advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_spatial_transcriptomics.html">Spatial transcriptomics</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_moca_100k.html">MOCA 100k cells</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_example_tsne2D_default.html">2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a07_example_tsne2D_advanced.html">Advanced mode on 2D t-SNE coordinates</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a04_example_spatial_transcriptomics_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of spatial transcriptomics with singleCellHaystack</h1>
                        <h4 class="author">Diego Diez</h4>
            
            <h4 class="date">2021-03-29</h4>
      
      
      <div class="hidden name"><code>a04_example_spatial_transcriptomics.Rmd</code></div>

    </div>

    
    
<p>We can apply <code>singleCellHaystack</code> to spatial transcriptomics data as well. Here we use <a href="https://github.com/satijalab/seurat">Seurat</a> (v3.2 or higher) and the spatial transcriptomics data available in the <a href="https://github.com/satijalab/seurat-data">SeuratData</a> package. For this example we use 10x Genomics Visium platform brain data. For more details about analyzing spatial transcriptomics with Seurat take a look at their spatial transcriptomics vignette <a href="https://satijalab.org/seurat/v3.2/spatial_vignette.html">here</a>.</p>
<div id="preparing-input-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preparing-input-data" class="anchor"></a>Preparing input data</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.satijalab.org/seurat">SeuratData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>We focus on the anterior1 slice.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"stxBrain"</span> <span class="op">%in%</span> <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu"><a href="http://www.satijalab.org/seurat/reference/InstalledData.html">InstalledData</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">"Dataset"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu"><a href="http://www.satijalab.org/seurat/reference/InstallData.html">InstallData</a></span><span class="op">(</span><span class="st">"stxBrain"</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">anterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://www.satijalab.org/seurat/reference/LoadData.html">LoadData</a></span><span class="op">(</span><span class="st">"stxBrain"</span>, type <span class="op">=</span> <span class="st">"anterior1"</span><span class="op">)</span>
<span class="va">anterior1</span></code></pre></div>
<pre><code>## An object of class Seurat 
## 31053 features across 2696 samples within 1 assay 
## Active assay: Spatial (31053 features, 0 variable features)</code></pre>
<p>We filter genes with less 10 cells with non-zero counts. This reduces the computational time by eliminating very lowly expressed genes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/AssayData.html">GetAssayData</a></span><span class="op">(</span><span class="va">anterior1</span>, slot <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span>
<span class="va">sel.ok</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">10</span>

<span class="va">anterior1</span> <span class="op">&lt;-</span> <span class="va">anterior1</span><span class="op">[</span><span class="va">sel.ok</span>, <span class="op">]</span>
<span class="va">anterior1</span></code></pre></div>
<pre><code>## An object of class Seurat 
## 12382 features across 2696 samples within 1 assay 
## Active assay: Spatial (12382 features, 0 variable features)</code></pre>
<p>We can plot the total number of counts per bead, superimposed on the image of the brain.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">anterior1</span>, features <span class="op">=</span> <span class="st">"nCount_Spatial"</span><span class="op">)</span></code></pre></div>
<p><img src="a04_example_spatial_transcriptomics_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
<p>We normalize the data we use log normalization.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">anterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="va">anterior1</span><span class="op">)</span></code></pre></div>
</div>
<div id="running-haystack-on-the-spatial-coordinates" class="section level1">
<h1 class="hasAnchor">
<a href="#running-haystack-on-the-spatial-coordinates" class="anchor"></a>Running <code>haystack</code> on the spatial coordinates</h1>
<p>At the moment there is no direct interface to apply <code>haystack</code> on spatial transcriptomics in Seurat. Once this version of Seurat is released we will add it. For now, we can obtain the two pieces of information required. One is the normalized counts and the other is the spatial coordinates. Then we pass to <code>haystack_2D</code> the coordinates of the beads and the detection matrix. Here we use a naive approach an define detected genes as those having non-zero counts.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/GetTissueCoordinates.html">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">anterior1</span>, <span class="st">"anterior1"</span><span class="op">)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/AssayData.html">GetAssayData</a></span><span class="op">(</span><span class="va">anterior1</span>, slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span>

<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/haystack_2D.html">haystack_2D</a></span><span class="op">(</span><span class="va">coord</span><span class="op">$</span><span class="va">imagecol</span>, <span class="va">coord</span><span class="op">$</span><span class="va">imagerow</span>, detection <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We can check the top genes with spatial biased distribution.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">sum</span>, n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></code></pre></div>
<pre><code>##                    D_KL log.p.vals  log.p.adj T.counts
## Slc17a7       0.7824319 -109.11039 -105.01760     1747
## Gpr88         0.9001888 -102.42973  -98.33694      748
## Pcp4l1        0.5545418  -99.91394  -95.82115     1397
## Nptxr         0.5639389  -99.26959  -95.17680     1321
## Pde10a        0.9010192  -99.20667  -95.11388      676
## Rgs9          1.0326093  -97.85555  -93.76276      590
## Penk          0.5558575  -97.71955  -93.62676     1291
## Tac1          0.7930455  -97.03810  -92.94531      769
## Lingo1        0.5455927  -96.62946  -92.53667     1284
## 3110035E14Rik 0.7236017  -94.49888  -90.40609      859
## Gng7          0.8888954  -93.83745  -89.74466      617
## Scn4b         0.9227326  -93.33386  -89.24107      596
## Cck           0.7235658  -91.90506  -87.81227     1955
## Ppp1r1b       0.5265363  -91.85167  -87.75888     1197
## Mobp          0.5411625  -91.42394  -87.33115     1750
## Stx1a         0.6118020  -90.73332  -86.64053     1002
## Adcy1         0.6142605  -90.04311  -85.95032      976
## Enc1          0.4797198  -89.91781  -85.82502     1653
## Hpca          0.5644309  -89.47208  -85.37929     1823
## Rasd2         0.7424966  -89.11496  -85.02217      659</code></pre>
<p>And we can visualize the expression of the 6 top-scoring genes in the spatial plot.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">anterior1</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sum</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="a04_example_spatial_transcriptomics_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
