<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of Slide-seqV2 data with singleCellHaystack â€¢ singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Analysis of Slide-seqV2 data with singleCellHaystack">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_scRNAseq.html">Example on scRNA-seq</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_spatial_visium.html">Example on Spatial Visium</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_spatial_slideseqV2.html">Example on Spatial Slide-seqV2</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_moca_100k.html">Application to MOCA 100k cells</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_pseudotime.html">Application to pseudotime</a>
    </li>
    <li>
      <a href="../../articles/examples/a07_gene_sets.html">Application to gene sets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of Slide-seqV2 data with
singleCellHaystack</h1>
                        <h4 data-toc-skip class="author">Alexis
Vandenbon</h4>
            
            <h4 data-toc-skip class="date">2022-12-23</h4>
      
      
      <div class="hidden name"><code>a04_example_spatial_slideseqV2.Rmd</code></div>

    </div>

    
    
<p>We can apply <code>singleCellHaystack</code> to spatial
transcriptomics data as well. Here we use <a href="https://github.com/satijalab/seurat" class="external-link">Seurat</a> and the spatial
transcriptomics data available in the <a href="https://github.com/satijalab/seurat-data" class="external-link">SeuratData</a> package.
For this example we use Slide-seqV2 hippocampus data. For more details
about analyzing spatial transcriptomics with Seurat take a look at their
spatial transcriptomics vignette <a href="https://satijalab.org/seurat/" class="external-link">here</a>.</p>
<div class="section level2">
<h2 id="preparing-input-data">Preparing input data<a class="anchor" aria-label="anchor" href="#preparing-input-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.satijalab.org/seurat" class="external-link">SeuratData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span></span></code></pre></div>
<p>We focus on the ssHippo slice.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"ssHippo"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/InstalledData.html" class="external-link">InstalledData</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">"Dataset"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/InstallData.html" class="external-link">InstallData</a></span><span class="op">(</span><span class="st">"ssHippo"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">ssHippo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/LoadData.html" class="external-link">LoadData</a></span><span class="op">(</span><span class="st">"ssHippo"</span><span class="op">)</span></span>
<span><span class="va">ssHippo</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 23264 features across 53173 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: Spatial (23264 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  1 image present: image</span></span></code></pre></div>
<p>We filter genes with less 10 cells with non-zero counts. This reduces
the computational time by eliminating very lowly expressed genes.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">ssHippo</span>, slot <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="va">sel.ok</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">10</span></span>
<span></span>
<span><span class="va">ssHippo</span> <span class="op">&lt;-</span> <span class="va">ssHippo</span><span class="op">[</span><span class="va">sel.ok</span>, <span class="op">]</span></span>
<span><span class="va">ssHippo</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 11390 features across 53173 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: Spatial (11390 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  1 image present: image</span></span></code></pre></div>
<p>We can plot the total number of counts per bead, superimposed on the
image of the brain.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">ssHippo</span>, features <span class="op">=</span> <span class="st">"nCount_Spatial"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a04_example_spatial_slideseqV2_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
<p>We normalize the data we use log normalization.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ssHippo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">ssHippo</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-haystack-on-the-spatial-coordinates">Running <code>haystack</code> on the spatial coordinates<a class="anchor" aria-label="anchor" href="#running-haystack-on-the-spatial-coordinates"></a>
</h2>
<p>The two inputs to <code>singleCellHaystack</code> are 1) the gene
expression data and 2) the spatial coordinates of the Slide-seq V2
pucks. Please note that we are not using an embedding as input space
here, but the actual 2D coordinates of spots inside the tissue. Since
this dataset contains about 53k spots and 11k genes, a run of
<code>singleCellHaystack</code> can take a few minutes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat.expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">ssHippo</span>, slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="va">dat.coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/GetTissueCoordinates.html" class="external-link">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">ssHippo</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/haystack.html">haystack</a></span><span class="op">(</span><span class="va">dat.coord</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, <span class="va">dat.expression</span><span class="op">)</span></span>
<span><span class="co">#&gt; ### calling haystack_continuous_highD()...</span></span>
<span><span class="co">#&gt; ### Using package sparseMatrixStats to speed up statistics in sparse matrices.</span></span>
<span><span class="co">#&gt; ### Calculating row-wise mean and SD...</span></span>
<span><span class="co">#&gt; ### Filtered 0 genes with zero variance...</span></span>
<span><span class="co">#&gt; ### Using 100 randomizations...</span></span>
<span><span class="co">#&gt; ### Using 100 genes to randomize...</span></span>
<span><span class="co">#&gt; ### converting expression data from dgCMatrix to dgRMatrix</span></span>
<span><span class="co">#&gt; ### scaling input data...</span></span>
<span><span class="co">#&gt; ### deciding grid points...</span></span>
<span><span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span></span>
<span><span class="co">#&gt; ### performing randomizations...</span></span>
<span><span class="co">#&gt; ### estimating p-values...</span></span>
<span><span class="co">#&gt; ### picking model for mean D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.014</span></span>
<span><span class="co">#&gt; ### best df    : 5</span></span>
<span><span class="co">#&gt; ### picking model for stdev D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.011</span></span>
<span><span class="co">#&gt; ### best df    : 4</span></span>
<span><span class="co">#&gt; ### returning result...</span></span></code></pre></div>
<p>We can check the top genes with spatial biased distribution.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt;              D_KL log.p.vals log.p.adj</span></span>
<span><span class="co">#&gt; PCP4   0.14476979  -262.8622 -258.8057</span></span>
<span><span class="co">#&gt; DDN    0.31739249  -254.4627 -250.4061</span></span>
<span><span class="co">#&gt; NRGN   0.24028799  -249.2989 -245.2424</span></span>
<span><span class="co">#&gt; MBP    0.09794532  -239.7026 -235.6461</span></span>
<span><span class="co">#&gt; CAMK2A 0.12320361  -234.3899 -230.3333</span></span>
<span><span class="co">#&gt; HPCA   0.15498142  -227.5796 -223.5231</span></span>
<span><span class="co">#&gt; TTR    0.23199941  -225.9053 -221.8488</span></span>
<span><span class="co">#&gt; ZIC1   0.69082139  -222.6275 -218.5709</span></span>
<span><span class="co">#&gt; PLP1   0.09916045  -220.5324 -216.4759</span></span>
<span><span class="co">#&gt; PRKCD  0.84449956  -219.1199 -215.0634</span></span></code></pre></div>
<p>And we can visualize the expression of the 6 top-scoring genes in the
spatial plot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top6</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res</span>, n <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">ssHippo</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">top6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="a04_example_spatial_slideseqV2_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
