<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application of 'advanced' mode on 2D t-SNE coordinates • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Application of 'advanced' mode on 2D t-SNE coordinates">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Application on toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_highD_default.html">Application on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_highD_advanced.html">Application of the advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_tsne2D_default.html">Application on 2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_example_tsne2D_advanced.html">Application of the advanced mode on 2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_example_spatial_transcriptomics.html">Application to spatial transcriptomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a05_example_tsne2D_advanced_files/header-attrs-2.2/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Application of ‘advanced’ mode on 2D t-SNE coordinates</h1>
                        <h4 class="author">Alexis Vandenbon</h4>
            
            <h4 class="date">2020-06-04</h4>
      
      
      <div class="hidden name"><code>a05_example_tsne2D_advanced.Rmd</code></div>

    </div>

    
    
<div id="tldr" class="section level1">
<h1 class="hasAnchor">
<a href="#tldr" class="anchor"></a>tl;dr</h1>
<p>You can indicate the general expression levels of cells using input parameter <code>use.advanced.sampling</code>. If you do that,<code>singleCellHaystack</code> takes the general expression levels into account and finds genes (DEGs) that are different from that general expression pattern.</p>
</div>
<div id="preparing-the-input-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preparing-the-input-data" class="anchor"></a>Preparing the input data</h1>
<p>The data used in these examples can be found <a href="https://genomics.virus.kyoto-u.ac.jp/alexisvdb/singleCellHaystack/">here</a>. We recommend downloading the .rda file and loading it using the <code><a href="https://rdrr.io/r/base/load.html">load()</a></code> function. You can also download the individual data files separately.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span>(<span class="st">"https://genomics.virus.kyoto-u.ac.jp/alexisvdb/singleCellHaystack/tabula_muris_marrow_P7_2.rda"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span>()
<span class="co">#&gt; [1] "dat.expression" "dat.pca"        "dat.tsne"       "dat.umap"</span></pre></body></html></div>
<p>This data should include the following objects:</p>
<ul>
<li>
<code>dat.expression</code>: a matrix object with the expression of genes (rows) in each cell (columns).</li>
<li>
<code>dat.pca</code>: the output of PCA. This data.frame contains thefirst 50 pricipal components.</li>
<li>
<code>dat.tsne</code>: a data.frame with t-SNE coordinates (2D).</li>
<li>
<code>dat.umap</code>: a data.frame with UMAP coordinates (2D).</li>
</ul>
<p>Let’s have a look at this example dataset:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co"># this data contains 12,030 genes and 1981 cells</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">dat.expression</span>)
<span class="co">#&gt; [1] 12030  1981</span>

<span class="co"># visualizing the cells in a t-SNE plot:</span>
<span class="fu">ggplot</span>(<span class="no">dat.tsne</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">V1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">V2</span>)) + <span class="fu">labs</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"t-SNE1"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"t-SNE2"</span>) + <span class="fu">geom_point</span>()</pre></body></html></div>
<p><img src="a05_example_tsne2D_advanced_files/figure-html/example1-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb3"><html><body><pre class="r">
<span class="co"># the t-SNE coordinates are based on the 50 PCs</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">dat.pca</span>) <span class="co"># the PCA data contains 1981 cells and 50 PCs</span>
<span class="co">#&gt; [1] 1981   50</span></pre></body></html></div>
<p>There are several groups of cells, although the borders between them are not clear and several might consist of additional subclusters. We can use <code>singleCellHaystack</code> to predict DEGs without relying on grouping of the cells into arbitrary clusters.</p>
<p><code>singleCellHaystack</code> has two required input parameters: 1) the coordinates of cells in the input space, and 2) a table showing which genes are detected in which cells. The definition of detection is left to the user. In this example we will use the median normalized read count of each gene as threshold for defining detection. Alternatively, we could also define genes with counts &gt; 0 as being detected, for example.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">median.per.gene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">dat.expression</span>,<span class="fl">1</span>,<span class="no">median</span>) <span class="co"># get the median read count per gene</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">median.per.gene</span>) <span class="co"># for many genes the median read count is 0</span>
<span class="co">#&gt;  Mrpl15  Lypla1  Rb1cc1    Sgk3   Snhg6   Tram1 </span>
<span class="co">#&gt;   0.000   0.000   0.000   0.000   0.000 136.243</span>
<span class="no">dat.detection</span> <span class="kw">&lt;-</span> <span class="no">dat.expression</span> <span class="kw">&gt;</span> <span class="no">median.per.gene</span> <span class="co"># use the medians as threshold for defining detection</span>
<span class="no">dat.detection</span>[<span class="fl">1</span>:<span class="fl">5</span>,<span class="fl">1</span>:<span class="fl">5</span>] <span class="co"># TRUE means detected, FALSE means not detected</span>
<span class="co">#&gt;        AAACCTGCAGTAACGG.1 AAACGGGAGAAGAAGC.1 AAACGGGAGACCGGAT.1</span>
<span class="co">#&gt; Mrpl15               TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Lypla1               TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Rb1cc1               TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Sgk3                 TRUE              FALSE              FALSE</span>
<span class="co">#&gt; Snhg6                TRUE              FALSE              FALSE</span>
<span class="co">#&gt;        AAACGGGAGCGCTCCA.1 AAACGGGAGGACGAAA.1</span>
<span class="co">#&gt; Mrpl15              FALSE               TRUE</span>
<span class="co">#&gt; Lypla1              FALSE               TRUE</span>
<span class="co">#&gt; Rb1cc1              FALSE              FALSE</span>
<span class="co">#&gt; Sgk3                FALSE              FALSE</span>
<span class="co">#&gt; Snhg6                TRUE              FALSE</span></pre></body></html></div>
<p>Now, that we have defined gene detection, let’s have a look at the t-SNE plot again, colouring cells by the number of genes detected in each cell:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">general.detection</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">dat.detection</span>, <span class="fl">2</span>, <span class="no">sum</span>)
<span class="fu">ggplot</span>(<span class="no">dat.tsne</span>, <span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">V1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">V2</span>, <span class="kw">colour</span> <span class="kw">=</span> <span class="no">general.detection</span>)) + <span class="fu">labs</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"t-SNE1"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"t-SNE2"</span>) +
  <span class="fu">geom_point</span>(<span class="kw">size</span><span class="kw">=</span><span class="fl">2</span>) + <span class="fu">scale_color_gradient</span>(<span class="kw">low</span><span class="kw">=</span><span class="st">"dodgerblue"</span>, <span class="kw">high</span><span class="kw">=</span><span class="st">"maroon2"</span>) + <span class="fu">labs</span>(<span class="kw">color</span> <span class="kw">=</span> <span class="st">"Det. genes"</span>)</pre></body></html></div>
<p><img src="a05_example_tsne2D_advanced_files/figure-html/example3-1.png" width="576" style="display: block; margin: auto;"></p>
<p>As can be seen in the last plot, cells in the bottom right have in general fewer detected genes. If a gene is detected predominantly in these cells and not in others, this tendency goes against the general trend in expression. Such a trend could be of particular interest.</p>
<p>We can use <code>singleCellHaystack</code> to predict DEGs without relying on grouping of the cells into arbitrary clusters, taking also into account general gene expression trends, using the option <code>use.advanced.sampling</code>.</p>
</div>
<div id="running-haystack-with-use-advanced-sampling" class="section level1">
<h1 class="hasAnchor">
<a href="#running-haystack-with-use-advanced-sampling" class="anchor"></a>Running <code>haystack</code> with <code>use.advanced.sampling</code>
</h1>
<p>First, load the package.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">singleCellHaystack</span>)</pre></body></html></div>
<p>Next, run <code>haystack</code> on the t-SNE coordinates. These are 2D coordinates, so we set ‘method’ to ‘2D’. We also give the detection values as input to ‘detection’. <code>use.advanced.sampling</code> is set to the vector of detected genes per cell, <code>general.detection</code>. By doing this, <code>haystack</code> will calculate cell distributions and perform randomizations taking into account the general detection levels. We also set a random seed to ensure replicability.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">123</span>)
<span class="no">res.tsne.adv</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../../reference/haystack.html">haystack</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">dat.tsne</span>, <span class="kw">detection</span> <span class="kw">=</span> <span class="no">dat.detection</span>, <span class="kw">use.advanced.sampling</span> <span class="kw">=</span> <span class="no">general.detection</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"2D"</span>)
<span class="co">#&gt; ### calling haystack_2D()...</span>
<span class="co">#&gt; ### setting parameters...</span>
<span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span>
<span class="co">#&gt; ### ... 1000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 2000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 3000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 4000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 5000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 6000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 7000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 8000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 9000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 10000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 11000 rows out of 12030 done</span>
<span class="co">#&gt; ### ... 12000 rows out of 12030 done</span>
<span class="co">#&gt; ### starting randomizations...</span>
<span class="co">#&gt; ### ... 10 sets out of 200 done</span>
<span class="co">#&gt; ### ... 20 sets out of 200 done</span>
<span class="co">#&gt; ### ... 30 sets out of 200 done</span>
<span class="co">#&gt; ### ... 40 sets out of 200 done</span>
<span class="co">#&gt; ### ... 50 sets out of 200 done</span>
<span class="co">#&gt; ### ... 60 sets out of 200 done</span>
<span class="co">#&gt; ### ... 70 sets out of 200 done</span>
<span class="co">#&gt; ### ... 80 sets out of 200 done</span>
<span class="co">#&gt; ### ... 90 sets out of 200 done</span>
<span class="co">#&gt; ### ... 100 sets out of 200 done</span>
<span class="co">#&gt; ### ... 110 sets out of 200 done</span>
<span class="co">#&gt; ### ... 120 sets out of 200 done</span>
<span class="co">#&gt; ### ... 130 sets out of 200 done</span>
<span class="co">#&gt; ### ... 140 sets out of 200 done</span>
<span class="co">#&gt; ### ... 150 sets out of 200 done</span>
<span class="co">#&gt; ### ... 160 sets out of 200 done</span>
<span class="co">#&gt; ### ... 170 sets out of 200 done</span>
<span class="co">#&gt; ### ... 180 sets out of 200 done</span>
<span class="co">#&gt; ### ... 190 sets out of 200 done</span>
<span class="co">#&gt; ### ... 200 sets out of 200 done</span>
<span class="co">#&gt; ### estimating p-values...</span>
<span class="co">#&gt; ### returning result...</span></pre></body></html></div>
<p>Let’s have a look at the most significant DEGs. The gene with the strongest differential expression is Trim10. We can plot the expression and detection of this gene using the <code>plot_gene_haystack</code> function.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span>(<span class="kw">res.haystack</span> <span class="kw">=</span> <span class="no">res.tsne.adv</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fl">5</span>)
<span class="co">#&gt;            D_KL log.p.vals log.p.adj T.counts</span>
<span class="co">#&gt; Trim10 2.698981  -3028.537 -3024.456      368</span>
<span class="co">#&gt; Gypa   2.400277  -2718.200 -2714.120      448</span>
<span class="co">#&gt; Slc4a1 2.342049  -2630.859 -2626.779      372</span>
<span class="co">#&gt; Snca   2.366295  -2552.161 -2548.081      474</span>
<span class="co">#&gt; Rsad2  2.162624  -2456.165 -2452.084      434</span>
<span class="co"># plotting detection of this gene</span>
<span class="fu"><a href="../../reference/plot_gene_haystack.html">plot_gene_haystack</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">dat.tsne</span>, <span class="kw">gene</span> <span class="kw">=</span> <span class="st">"Trim10"</span>, <span class="kw">expression</span> <span class="kw">=</span> <span class="no">dat.detection</span>)</pre></body></html></div>
<p><img src="a05_example_tsne2D_advanced_files/figure-html/example6-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co"># plotting log expression of this gene</span>
<span class="fu"><a href="../../reference/plot_gene_haystack.html">plot_gene_haystack</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">dat.tsne</span>, <span class="kw">gene</span> <span class="kw">=</span> <span class="st">"Trim10"</span>, <span class="kw">expression</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">dat.expression</span>))</pre></body></html></div>
<p><img src="a05_example_tsne2D_advanced_files/figure-html/example6-2.png" width="576" style="display: block; margin: auto;"></p>
<p>From these plots, we can see that Trim10 is expressed almost exclusively in the bottom right group of cells. As seen above, these cells have in general fewer detected genes, so the pattern of Trim10 is especially surprizing: it is deteced in cells that express in general few genes.</p>
<p>This explains why the advanced mode of <code>singleCellHaystack</code> judges Trim10 to be the most significant DEG.</p>
</div>
<div id="clustering-and-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering-and-visualization" class="anchor"></a>Clustering and visualization</h1>
<p>Next, let’s take the top 1000 DEGs, and cluster them by their expression pattern in the 2D t-SNE coordinates. Here we use <code>hclust_haystack</code>, which uses hierarchical clustering. Alternatively, we could use <code>kmeans_haystack</code> for k-means clustering.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co"># get the top 1000 DEGs in the result</span>
<span class="no">res.top</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span>(<span class="kw">res.haystack</span> <span class="kw">=</span> <span class="no">res.tsne.adv</span>, <span class="kw">n</span> <span class="kw">=</span> <span class="fl">1000</span>)
<span class="co"># cluster DEGs by their expression pattern in the 2D plot</span>
<span class="no">genes.top</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(<span class="no">res.top</span>)
<span class="no">res.hc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../../reference/hclust_haystack.html">hclust_haystack</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">dat.tsne</span>, <span class="kw">genes</span> <span class="kw">=</span> <span class="no">genes.top</span>, <span class="kw">detection</span> <span class="kw">=</span> <span class="no">dat.detection</span>)</pre></body></html></div>
<p><code>hclust_haystack</code> returns as result a <code>hclust</code> tree, which we can cut into clusters using the <code>cutree</code> function. Here, we arbitrarily set the number of clusters to 5.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">res.hc.clusters</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span>(<span class="no">res.hc</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">5</span>)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">res.hc.clusters</span>)
<span class="co">#&gt; res.hc.clusters</span>
<span class="co">#&gt;   1   2   3   4   5 </span>
<span class="co">#&gt; 217 221 136 294 132</span></pre></body></html></div>
<p>Let’s run through the 5 clusters and plot their averaged detection pattern using <code>plot_gene_set_haystack</code>, which is similar to <code>plot_gene_haystack</code> but uses a set of genes as input instead of just 1 gene.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">pl</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="fl">1</span>:<span class="fl">5</span>, <span class="kw">function</span>(<span class="no">cluster</span>) {
  <span class="no">gene.set</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">res.hc.clusters</span>)[<span class="no">res.hc.clusters</span><span class="kw">==</span><span class="no">cluster</span>]
  <span class="no">plot.title</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"Cluster "</span>, <span class="no">cluster</span>)
  <span class="no">p</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../../reference/plot_gene_set_haystack.html">plot_gene_set_haystack</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">dat.tsne</span>, <span class="kw">genes</span> <span class="kw">=</span> <span class="no">gene.set</span>, <span class="kw">detection</span> <span class="kw">=</span> <span class="no">dat.detection</span>)
  <span class="no">p</span> + <span class="fu">ggtitle</span>(<span class="no">plot.title</span>) + <span class="fu">theme</span>(<span class="kw">legend.title</span> <span class="kw">=</span> <span class="fu">element_text</span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">8</span>))
})
<span class="fu">plot_grid</span>(<span class="kw">plotlist</span> <span class="kw">=</span> <span class="no">pl</span>, <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<p><img src="a05_example_tsne2D_advanced_files/figure-html/example9-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">res.hc.clusters</span>[<span class="st">"Trim10"</span>] <span class="co"># the most significant DEG is in cluster 5</span>
<span class="co">#&gt; Trim10 </span>
<span class="co">#&gt;      5</span></pre></body></html></div>
<p>The most significant DEG, Trim10, was clustered into cluster 5. Comparing its expression pattern (see above) with that of each cluster, we can indeed see that it fits most closely with that of cluster 5. Cluster 5 genes are on average detected almost exclusively in the bottom right group of cells.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
