<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of gene set activities with singleCellHaystack • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Analysis of gene set activities with singleCellHaystack">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_highD_default.html">Multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_highD_advanced.html">Advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_spatial_transcriptomics.html">Spatial transcriptomics</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_moca_100k.html">MOCA 100k cells</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_example_tsne2D_default.html">2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../../articles/examples/a07_example_tsne2D_advanced.html">Advanced mode on 2D t-SNE coordinates</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of gene set activities with
singleCellHaystack</h1>
                        <h4 data-toc-skip class="author">Alexis
Vandenbon</h4>
            
            <h4 data-toc-skip class="date">2022-11-07</h4>
      
      
      <div class="hidden name"><code>a07_gene_sets.Rmd</code></div>

    </div>

    
    
<p>Because <code>singleCellHaystack</code> makes few assumptions about
the properties of input data, it can be applied to any numerical data.
Here we used <a href="https://satijalab.org/seurat/reference/addmodulescore" class="external-link">Seurat’s
AddModuleScore</a> to calculated scores for sets of genes that share a
common function (based on GO annotations), and applied it to a spatial
transcriptomics dataset. For this example we use 10x Genomics Visium
platform brain data.</p>
<div class="section level2">
<h2 id="preparing-input-data">Preparing input data<a class="anchor" aria-label="anchor" href="#preparing-input-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.satijalab.org/seurat" class="external-link">SeuratData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span></span></code></pre></div>
<p>We focus on the 10x Genomics Visium anterior1 slice. This datasets
contains 31,053 genes and 2,696 spots.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"stxBrain"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/InstalledData.html" class="external-link">InstalledData</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">"Dataset"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">SeuratData</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/InstallData.html" class="external-link">InstallData</a></span><span class="op">(</span><span class="st">"stxBrain"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">anterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/LoadData.html" class="external-link">LoadData</a></span><span class="op">(</span><span class="st">"stxBrain"</span>, type <span class="op">=</span> <span class="st">"anterior1"</span><span class="op">)</span></span>
<span><span class="va">anterior1</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 31053 features across 2696 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: Spatial (31053 features, 0 variable features)</span></span>
<span><span class="co">#&gt;  1 image present: anterior1</span></span></code></pre></div>
<p>We used Seurat’s AddModuleScore function to calculate numerical
values reflecting the average activity of sets of genes associated with
common GO terms. We did this for 2,939 Biological Process GO terms that
have between 20 and 250 genes in this anterior1 dataset. This
calculation takes some time. Here we will use pre-calculated scores that
we made <a href="https://figshare.com/articles/dataset/Example_datasets_for_singleCellHaystack_GitHub_vignettes_2022_/21499071" class="external-link">available
on figshare</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://figshare.com/ndownloader/files/38105937"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/ls.html" class="external-link">ls</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "anterior1"  "dat.scores"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dat.scores</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2939 2696</span></span>
<span><span class="va">dat.scores</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                   AAACAAGTATCTCCCA-1 AAACACCAATAACTGC-1</span></span>
<span><span class="co">#&gt; GOBP_ACETYL_COA_METABOLIC_PROCESS             0.5663             0.4681</span></span>
<span><span class="co">#&gt; GOBP_ACIDIC_AMINO_ACID_TRANSPORT              0.3746             0.8791</span></span>
<span><span class="co">#&gt; GOBP_ACID_SECRETION                           0.4978             0.5679</span></span>
<span><span class="co">#&gt;                                   AAACAGAGCGACTCCT-1</span></span>
<span><span class="co">#&gt; GOBP_ACETYL_COA_METABOLIC_PROCESS             0.4743</span></span>
<span><span class="co">#&gt; GOBP_ACIDIC_AMINO_ACID_TRANSPORT              0.3768</span></span>
<span><span class="co">#&gt; GOBP_ACID_SECRETION                           0.4507</span></span></code></pre></div>
<p>We can add these module scores as meta data to the Seurat object, and
visualize one of them as an example. Here we visualize the scores for
the GO term “acetyl CoA metabolic process”.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anterior1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/AddMetaData.html" class="external-link">AddMetaData</a></span><span class="op">(</span><span class="va">anterior1</span>,<span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">dat.scores</span><span class="op">)</span>, col.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat.scores</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">anterior1</span>, features <span class="op">=</span> <span class="st">"GOBP_ACETYL_COA_METABOLIC_PROCESS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a07_gene_sets_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="running-haystack-on-the-spatial-coordinates">Running <code>haystack</code> on the spatial coordinates<a class="anchor" aria-label="anchor" href="#running-haystack-on-the-spatial-coordinates"></a>
</h2>
<p>Next, we want to find biological processes that have a biased
activity within the tissue. For this, the two inputs to
<code>singleCellHaystack</code> are 1) the module score data and 2) the
spatial coordinates of the Visium spots. Please note that we are not
using an embedding as input space here, but the actual 2D coordinates of
spots inside the tissue. Since this Visium dataset only contains about
2.7k spots and we have scores for about 2.9k GO terms, running
<code>singleCellHaystack</code> should only take about a minute.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat.coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/GetTissueCoordinates.html" class="external-link">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">anterior1</span>, <span class="st">"anterior1"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/haystack.html">haystack</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat.coord</span>, expression <span class="op">=</span> <span class="va">dat.scores</span><span class="op">)</span></span>
<span><span class="co">#&gt; ### calling haystack_continuous_highD()...</span></span>
<span><span class="co">#&gt; ### Using package sparseMatrixStats to speed up statistics in sparse matrices.</span></span>
<span><span class="co">#&gt; ### Calculating row-wise mean and SD...</span></span>
<span><span class="co">#&gt; ### Filtered 0 genes with zero variance...</span></span>
<span><span class="co">#&gt; ### Using 100 randomizations...</span></span>
<span><span class="co">#&gt; ### Using 100 genes to randomize...</span></span>
<span><span class="co">#&gt; ### scaling input data...</span></span>
<span><span class="co">#&gt; ### deciding grid points...</span></span>
<span><span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span></span>
<span><span class="co">#&gt; ### performing randomizations...</span></span>
<span><span class="co">#&gt; ### estimating p-values...</span></span>
<span><span class="co">#&gt; ### picking model for mean D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.013</span></span>
<span><span class="co">#&gt; ### best df    : 1</span></span>
<span><span class="co">#&gt; ### picking model for stdev D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.01</span></span>
<span><span class="co">#&gt; ### best df    : 1</span></span>
<span><span class="co">#&gt; ### returning result...</span></span></code></pre></div>
<p>We can check the top GO processes with spatially biased score
distributions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                                     D_KL log.p.vals log.p.adj</span></span>
<span><span class="co">#&gt; GOBP_NEUROTRANSMITTER_SECRETION               0.02094824  -116.1739 -112.7057</span></span>
<span><span class="co">#&gt; GOBP_NEUROTRANSMITTER_TRANSPORT               0.01724916  -115.0818 -111.6136</span></span>
<span><span class="co">#&gt; GOBP_REGULATION_OF_NEUROTRANSMITTER_TRANSPORT 0.01639296  -114.8745 -111.4063</span></span>
<span><span class="co">#&gt; GOBP_LOCOMOTORY_BEHAVIOR                      0.02117002  -113.0179 -109.5497</span></span>
<span><span class="co">#&gt; GOBP_SYNAPTIC_VESICLE_EXOCYTOSIS              0.01941167  -112.9365 -109.4683</span></span>
<span><span class="co">#&gt; GOBP_RESPONSE_TO_AMPHETAMINE                  0.06370640  -112.7320 -109.2638</span></span>
<span><span class="co">#&gt; GOBP_RESPONSE_TO_ISOQUINOLINE_ALKALOID        0.04415398  -112.5536 -109.0854</span></span>
<span><span class="co">#&gt; GOBP_REGULATION_OF_SYNAPTIC_PLASTICITY        0.01740920  -112.0729 -108.6047</span></span>
<span><span class="co">#&gt; GOBP_REGULATED_EXOCYTOSIS                     0.01826615  -111.7330 -108.2648</span></span>
<span><span class="co">#&gt; GOBP_DENDRITE_DEVELOPMENT                     0.01639215  -111.1884 -107.7202</span></span></code></pre></div>
<p>And we can visualize the distribution of scores for the 6 top-scoring
GO terms in the spatial plot as an example.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top6</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res</span>, n <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">anterior1</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">top6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="a07_gene_sets_files/figure-html/unnamed-chunk-7-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
