<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Clustering singleCellHaystack results into gene modules • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Clustering singleCellHaystack results into gene modules">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../../articles/examples/a02_example_scRNAseq.html">Example on scRNA-seq</a>
    </li>
    <li>
      <a href="../../articles/examples/a03_example_spatial_visium.html">Example on Spatial Visium</a>
    </li>
    <li>
      <a href="../../articles/examples/a04_example_spatial_slideseqV2.html">Example on Spatial Slide-seqV2</a>
    </li>
    <li>
      <a href="../../articles/examples/a05_moca_100k.html">Application to MOCA 100k cells</a>
    </li>
    <li>
      <a href="../../articles/examples/a06_pseudotime.html">Application to pseudotime</a>
    </li>
    <li>
      <a href="../../articles/examples/a07_gene_sets.html">Application to gene sets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Clustering singleCellHaystack results into gene
modules</h1>
                        <h4 data-toc-skip class="author">Diego Diez</h4>
            
            <h4 data-toc-skip class="date">2022-10-09</h4>
      
      
      <div class="hidden name"><code>a08_gene_clustering.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Attaching SeuratObject</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.satijalab.org/seurat" class="external-link">SeuratData</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ── <span style="font-weight: bold;">Installed datasets</span> ───────────────────────────────────── SeuratData v0.2.2 ──</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">ifnb    </span> 3.1.0                        <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">ssHippo </span> 3.1.4</span></span>
<span><span class="co">## <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">pbmc3k  </span> 3.1.4                        <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">stxBrain</span> 0.1.1</span></span></code></pre>
<pre><code><span><span class="co">## ────────────────────────────────────── Key ─────────────────────────────────────</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> Dataset loaded successfully</span></span>
<span><span class="co">## <span style="color: #BBBB00;">❯</span> Dataset built with a newer version of Seurat than installed</span></span>
<span><span class="co">## <span style="color: #BB0000;">❓</span> Unknown version of Seurat installed</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: grid</span></span></code></pre>
<pre><code><span><span class="co">## ========================================</span></span>
<span><span class="co">## ComplexHeatmap version 2.14.0</span></span>
<span><span class="co">## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span></span>
<span><span class="co">## Github page: https://github.com/jokergoo/ComplexHeatmap</span></span>
<span><span class="co">## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use it in published research, please cite either one:</span></span>
<span><span class="co">## - Gu, Z. Complex Heatmap Visualization. iMeta 2022.</span></span>
<span><span class="co">## - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span></span>
<span><span class="co">##     genomic data. Bioinformatics 2016.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The new InteractiveComplexHeatmap package can directly export static </span></span>
<span><span class="co">## complex heatmaps into an interactive Shiny app with zero effort. Have a try!</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## This message can be suppressed by:</span></span>
<span><span class="co">##   suppressPackageStartupMessages(library(ComplexHeatmap))</span></span>
<span><span class="co">## ========================================</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">select_cell_markers</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">scores</span>, <span class="va">max.overlap</span><span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">scores</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">pos</span> <span class="op">&lt;-</span> <span class="va">x</span> <span class="op">&gt;</span> <span class="fl">0</span></span>
<span>    <span class="va">npos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">npos</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">&amp;</span> <span class="va">npos</span> <span class="op">&lt;=</span> <span class="va">max.overlap</span><span class="op">)</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="co">#rownames(scores)[sel]</span></span>
<span>  <span class="va">scores</span><span class="op">[</span><span class="va">sel</span>,<span class="op">]</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this vignette we use the PBMC dataset from 10x available in the
<code>SeuratData</code> package.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/LoadData.html" class="external-link">LoadData</a></span><span class="op">(</span><span class="st">"pbmc3k"</span>, <span class="st">"pbmc3k.final"</span><span class="op">)</span></span>
<span><span class="va">pbmc</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 13714 features across 2638 samples within 1 assay </span></span>
<span><span class="co">## Active assay: RNA (13714 features, 2000 variable features)</span></span>
<span><span class="co">##  2 dimensional reductions calculated: pca, umap</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">pbmc</span>, label<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoAxes</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="run-haystack">Run haystack<a class="anchor" aria-label="anchor" href="#run-haystack"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Embeddings.html" class="external-link">Embeddings</a></span><span class="op">(</span><span class="va">pbmc</span>, reduction<span class="op">=</span><span class="st">"pca"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">coord</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2638   50</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">pbmc</span>, assay<span class="op">=</span><span class="st">"RNA"</span>, slot<span class="op">=</span><span class="st">"data"</span><span class="op">)</span></span>
<span><span class="va">exprs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## 4 x 4 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##               AAACATACAACCAC AAACATTGAGCTAC AAACATTGATCAGC AAACCGTGCTTCCG</span></span>
<span><span class="co">## AL627309.1                 .              .              .              .</span></span>
<span><span class="co">## AP006222.2                 .              .              .              .</span></span>
<span><span class="co">## RP11-206L10.2              .              .              .              .</span></span>
<span><span class="co">## RP11-206L10.9              .              .              .              .</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/haystack_continuous_highD.html">haystack_continuous_highD</a></span><span class="op">(</span><span class="va">coord</span>, <span class="va">exprs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ### calling haystack_continuous_highD()...</span></span></code></pre>
<pre><code><span><span class="co">## ### Using package sparseMatrixStats to speed up statistics in sparse matrices.</span></span></code></pre>
<pre><code><span><span class="co">## ### Calculating row-wise mean and SD...</span></span></code></pre>
<pre><code><span><span class="co">## ### Filtered 1 genes with zero variance...</span></span></code></pre>
<pre><code><span><span class="co">## ### Using 100 randomizations...</span></span></code></pre>
<pre><code><span><span class="co">## ### Using 100 genes to randomize...</span></span></code></pre>
<pre><code><span><span class="co">## ### converting expression data from dgCMatrix to dgRMatrix</span></span></code></pre>
<pre><code><span><span class="co">## ### scaling input data...</span></span></code></pre>
<pre><code><span><span class="co">## ### deciding grid points...</span></span></code></pre>
<pre><code><span><span class="co">## ### calculating Kullback-Leibler divergences...</span></span></code></pre>
<pre><code><span><span class="co">## ### performing randomizations...</span></span></code></pre>
<pre><code><span><span class="co">## ### estimating p-values...</span></span></code></pre>
<pre><code><span><span class="co">## ### picking model for mean D_KL...</span></span></code></pre>
<pre><code><span><span class="co">## ### using natural splines</span></span></code></pre>
<pre><code><span><span class="co">## ### best RMSD  : 0.049</span></span></code></pre>
<pre><code><span><span class="co">## ### best df    : 10</span></span></code></pre>
<pre><code><span><span class="co">## ### picking model for stdev D_KL...</span></span></code></pre>
<pre><code><span><span class="co">## ### using natural splines</span></span></code></pre>
<pre><code><span><span class="co">## ### best RMSD  : 0.08</span></span></code></pre>
<pre><code><span><span class="co">## ### best df    : 9</span></span></code></pre>
<pre><code><span><span class="co">## ### returning result...</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/plot_rand_fit.html">plot_rand_fit</a></span><span class="op">(</span><span class="va">res</span>, <span class="st">"mean"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../../reference/plot_rand_fit.html">plot_rand_fit</a></span><span class="op">(</span><span class="va">res</span>, <span class="st">"sd"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-6-2.png" width="700"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sum</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   D_KL log.p.vals log.p.adj</span></span>
<span><span class="co">## LY6G6F     1.994737552  -61.62395 -57.48682</span></span>
<span><span class="co">## AP001189.4 0.560430261  -53.46206 -49.32493</span></span>
<span><span class="co">## GP9        0.369618675  -48.62653 -44.48940</span></span>
<span><span class="co">## GZMB       0.005030405  -47.99422 -43.85709</span></span>
<span><span class="co">## CLDN5      0.864521392  -44.88222 -40.74509</span></span>
<span><span class="co">## FGFBP2     0.004488329  -43.42574 -39.28861</span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">sum</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sum</span><span class="op">)</span><span class="op">)</span>, y<span class="op">=</span><span class="va">log.p.adj</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">1000</span><span class="op">)</span>, color<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"limegreen"</span>, <span class="st">"violetred"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Next we look at the top 100 genes identified by haystack.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_100_haystack</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sum</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">top_100_haystack</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   D_KL log.p.vals log.p.adj</span></span>
<span><span class="co">## LY6G6F     1.994737552  -61.62395 -57.48682</span></span>
<span><span class="co">## AP001189.4 0.560430261  -53.46206 -49.32493</span></span>
<span><span class="co">## GP9        0.369618675  -48.62653 -44.48940</span></span>
<span><span class="co">## GZMB       0.005030405  -47.99422 -43.85709</span></span>
<span><span class="co">## CLDN5      0.864521392  -44.88222 -40.74509</span></span>
<span><span class="co">## FGFBP2     0.004488329  -43.42574 -39.28861</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="using-whole-expression-profiles">Using whole expression profiles<a class="anchor" aria-label="anchor" href="#using-whole-expression-profiles"></a>
</h2>
<p>We can use the top genes to calculate gene clusters using the single
cell expression data.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exprs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">top_100_haystack</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="va">m</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            AAACATACAACCAC AAACATTGAGCTAC AAACATTGATCAGC</span></span>
<span><span class="co">## LY6G6F                  0              0              0</span></span>
<span><span class="co">## AP001189.4              0              0              0</span></span>
<span><span class="co">## GP9                     0              0              0</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">m</span>, column_split<span class="op">=</span><span class="va">pbmc</span><span class="op">$</span><span class="va">seurat_annotations</span>, name<span class="op">=</span><span class="st">"Expression"</span>, show_column_names<span class="op">=</span><span class="cn">FALSE</span>, column_title_rot<span class="op">=</span><span class="fl">45</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `use_raster` is automatically set to TRUE for a matrix with more than</span></span>
<span><span class="co">## 2000 columns You can control `use_raster` argument by explicitly</span></span>
<span><span class="co">## setting TRUE/FALSE to it.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Set `ht_opt$message = FALSE` to turn off this message.</span></span></code></pre>
<pre><code><span><span class="co">## 'magick' package is suggested to install to give better rasterization.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Set `ht_opt$message = FALSE` to turn off this message.</span></span></code></pre>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-11-1.png" width="384"></p>
</div>
<div class="section level2">
<h2 id="usingn-cluster-wise-averages">Usingn cluster-wise averages<a class="anchor" aria-label="anchor" href="#usingn-cluster-wise-averages"></a>
</h2>
<p>We can calculate cluster scores for each gene. Basically we compute
the average of each gene in each cluster.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/AverageExpression.html" class="external-link">AverageExpression</a></span><span class="op">(</span><span class="va">pbmc</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">top_100_haystack</span><span class="op">)</span>, assays<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            Naive CD4 T Memory CD4 T  CD14+ Mono         B    CD8 T FCGR3A+ Mono</span></span>
<span><span class="co">## LY6G6F       0.0000000   0.00000000 0.000000000 0.0000000 0.000000   0.00000000</span></span>
<span><span class="co">## AP001189.4   0.0000000   0.00000000 0.000000000 0.0000000 0.000000   0.03068193</span></span>
<span><span class="co">## GP9          0.0000000   0.01189881 0.008018989 0.0000000 0.000000   0.01159001</span></span>
<span><span class="co">## GZMB         0.1327085   0.16999619 0.198919462 0.1178744 5.011299   0.24145661</span></span>
<span><span class="co">## CLDN5        0.0000000   0.00000000 0.000000000 0.0000000 0.000000   0.01513322</span></span>
<span><span class="co">## FGFBP2       0.1065733   0.13085714 0.091726089 0.1935568 4.541959   0.12754242</span></span>
<span><span class="co">##                  NK        DC  Platelet</span></span>
<span><span class="co">## LY6G6F      0.00000 0.0000000  7.359751</span></span>
<span><span class="co">## AP001189.4  0.00000 0.0000000 12.269047</span></span>
<span><span class="co">## GP9         0.00000 0.0000000 33.938348</span></span>
<span><span class="co">## GZMB       49.85524 7.4516567  1.264483</span></span>
<span><span class="co">## CLDN5       0.00000 0.0000000 11.913162</span></span>
<span><span class="co">## FGFBP2     25.64858 0.1033588  0.000000</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes.pass</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">scores</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">genes.pass</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## genes.pass</span></span>
<span><span class="co">##  0  1  2  3  4  5  6 </span></span>
<span><span class="co">##  4 36 22 19 13  5  1</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">[</span><span class="va">genes.pass</span>, <span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>We need to scale the scores.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            Naive CD4 T Memory CD4 T CD14+ Mono          B      CD8 T</span></span>
<span><span class="co">## LY6G6F      -0.3333333   -0.3333333 -0.3333333 -0.3333333 -0.3333333</span></span>
<span><span class="co">## AP001189.4  -0.3342704   -0.3342704 -0.3342704 -0.3342704 -0.3342704</span></span>
<span><span class="co">## GP9         -0.3336815   -0.3326296 -0.3329726 -0.3336815 -0.3336815</span></span>
<span><span class="co">## GZMB        -0.4330890   -0.4307911 -0.4290087 -0.4340032 -0.1324408</span></span>
<span><span class="co">## CLDN5       -0.3338095   -0.3338095 -0.3338095 -0.3338095 -0.3338095</span></span>
<span><span class="co">## FGFBP2      -0.3939467   -0.3910753 -0.3957023 -0.3836615  0.1305074</span></span>
<span><span class="co">##            FCGR3A+ Mono         NK          DC   Platelet</span></span>
<span><span class="co">## LY6G6F       -0.3333333 -0.3333333 -0.33333333  2.6666667</span></span>
<span><span class="co">## AP001189.4   -0.3267658 -0.3342704 -0.33427038  2.6666585</span></span>
<span><span class="co">## GP9          -0.3326569 -0.3336815 -0.33368148  2.6666664</span></span>
<span><span class="co">## GZMB         -0.4263873  2.6311137  0.01794873 -0.3633423</span></span>
<span><span class="co">## CLDN5        -0.3299980 -0.3338095 -0.33380950  2.6666645</span></span>
<span><span class="co">## FGFBP2       -0.3914673  2.6262208 -0.39432680 -0.4065483</span></span></code></pre>
<p>We can see that there are clear cluster specific markers.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">scores</span>, name<span class="op">=</span><span class="st">"Score"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a08_gene_clustering_files/figure-html/unnamed-chunk-16-1.png" width="384">
# Using the distribution P</p>
<p>Another approach is to use the distribution P used by haystack. This
is the density weighted expression at the grid points. Haystack returns
by default the density.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">info</span><span class="op">$</span><span class="va">P</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
