<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Finding Needles (=differentially Expressed Genes) in Haystacks (=single Cell Data) • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Finding Needles (=differentially Expressed Genes) in Haystacks (=single Cell Data)">
<meta property="og:description" content="Identification of differentially expressed genes (DEGs) is a key step in single-cell 
    transcriptomics data analysis. singleCellHaystack predicts DEGs without relying on 
    clustering of cells into arbitrary clusters. Single-cell RNA-seq (scRNA-seq) data is 
    often processed to fewer dimensions using Principal Component Analysis (PCA) and 
    represented in 2-dimensional plots (e.g. t-SNE or UMAP plots). singleCellHaystack 
    uses Kullback-Leibler divergence to find genes that are expressed in subsets of cells 
    that are non-randomly positioned in a these multi-dimensional spaces or 2D representations. 
    For the theoretical background of singleCellHaystack we refer to 
    Vandenbon and Diez (Nature Communications, 2020) &lt;https://doi.org/10.1038/s41467-020-17900-3&gt;.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/a01_toy_example.html">Application on toy example</a>
    </li>
    <li>
      <a href="articles/examples/a02_example_highD_default.html">Application on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="articles/examples/a03_example_highD_advanced.html">Application of the advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="articles/examples/a04_example_tsne2D_default.html">Application on 2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="articles/examples/a05_example_tsne2D_advanced.html">Application of the advanced mode on 2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="articles/examples/a06_example_spatial_transcriptomics.html">Application to spatial transcriptomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="singlecellhaystack" class="section level2">
<h2 class="hasAnchor">
<a href="#singlecellhaystack" class="anchor"></a>singleCellHaystack</h2>
<!-- badges: start -->

<p><code>singleCellHaystack</code> is a package for predicting differentially expressed genes (DEGs) in single cell transcriptome data. It does so without relying on clustering of cells into arbitrary clusters! Single-cell RNA-seq (scRNA-seq) data is often processed to fewer dimensions using Principal Component Analysis (PCA) and represented in 2-dimensional plots (e.g. t-SNE or UMAP plots). <code>singleCellHaystack</code> uses Kullback-Leibler Divergence to find genes that are expressed in subsets of cells that are non-randomly positioned in a these multi-dimensional spaces or 2D representations.</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Our manuscript describing <code>singleCellHaystack</code> has been published in <a href="https://doi.org/10.1038/s41467-020-17900-3">Nature Communications</a>.</p>
<p>If you use <code>singleCellHaystack</code> in your research please cite our work using:</p>
<p>Vandenbon A, Diez D (2020). “A clustering-independent method for finding differentially expressed genes in single-cell transcriptome data.” <em>Nature Communications</em>, <em>11</em>(1), 4318. doi: 10.1038/s41467-020-17900-3 (URL: <a href="https://doi.org/10.1038/s41467-020-17900-3" class="uri">https://doi.org/10.1038/s41467-020-17900-3</a>).</p>
</div>
<div id="documentation-and-demo" class="section level2">
<h2 class="hasAnchor">
<a href="#documentation-and-demo" class="anchor"></a>Documentation and Demo</h2>
<p>Our <a href="https://alexisvdb.github.io/singleCellHaystack/">documentation</a> includes a few example applications showing how to use our package:</p>
<ul>
<li><a href="https://alexisvdb.github.io/singleCellHaystack/articles/a01_toy_example.html">Application on toy example</a></li>
<li><a href="https://alexisvdb.github.io/singleCellHaystack/articles/examples/a02_example_highD_default.html">Application on multi-dimensional coordinates</a></li>
<li><a href="https://alexisvdb.github.io/singleCellHaystack/articles/examples/a03_example_highD_advanced.html">Application of the advanced mode on multi-dimensional coordinates</a></li>
<li><a href="https://alexisvdb.github.io/singleCellHaystack/articles/examples/a04_example_tsne2D_default.html">Application on 2D t-SNE coordinates</a></li>
<li><a href="https://alexisvdb.github.io/singleCellHaystack/articles/examples/a05_example_tsne2D_advanced.html">Application of the advanced mode on 2D t-SNE coordinates</a></li>
<li><a href="https://alexisvdb.github.io/singleCellHaystack/articles/examples/a06_example_spatial_transcriptomics.html">Application to spatial transcriptomics</a></li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the released version of <code>singleCellHaystack</code> from <a href="https://CRAN.R-project.org/package=singleCellHaystack">CRAN</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"singleCellHaystack"</span>)</pre></div>
<p>You can also install <code>singleCellHaystack</code> from the GitHub repository as shown below. Typical installation times should be less than 1 minute.</p>
<div class="sourceCode" id="cb2"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">remotes</span>)
<span class="kw pkg">remotes</span><span class="kw ns">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span>(<span class="st">"alexisvdb/singleCellHaystack"</span>)</pre></div>
</div>
<div id="system-requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#system-requirements" class="anchor"></a>System Requirements</h2>
<div id="hardware-requirements" class="section level3">
<h3 class="hasAnchor">
<a href="#hardware-requirements" class="anchor"></a>Hardware Requirements</h3>
<p><code>singleCellHaystack</code> requires only a standard computer with sufficient RAM to support running R or RStudio. Memory requirements depend on the size of the input dataset.</p>
</div>
<div id="software-requirements" class="section level3">
<h3 class="hasAnchor">
<a href="#software-requirements" class="anchor"></a>Software Requirements</h3>
<p>This package has been tested on Windows (Windows 10), macOS (Mojave 10.14.1 and Catalina 10.15.1), and Linux (CentOS 6.9 and Ubuntu 19.10).</p>
<p><code>singleCellHaystack</code> depends on the following packages: splines (3.6.0), ggplot2 (3.2.0), reshape2 (1.4.3).</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=singleCellHaystack">https://​cloud.r-project.org/​package=singleCellHaystack</a>
</li>
<li>Learn more at <br><a href="https://doi.org/10.1038/s41467-020-17900-3">https://​doi.org/​10.1038/​s41467-020-17900-3</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing singleCellHaystack</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Alexis Vandenbon <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-2180-5732" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Diego Diez <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-2325-4893" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/alexisvdb/singleCellHaystack/actions"><img src="https://github.com/alexisvdb/singleCellHaystack/workflows/R-CMD-check/badge.svg" alt="R build status"></a></li>
<li><a href="https://cran.r-project.org/package=singleCellHaystack"><img src="https://www.r-pkg.org/badges/version/singleCellHaystack" alt="CRAN_Status_Badge"></a></li>
<li><a href="https://cran.r-project.org/package=singleCellHaystack"><img src="https://cranlogs.r-pkg.org/badges/singleCellHaystack" alt="CRAN Downloads"></a></li>
<li><a href="https://cran.r-project.org/package=singleCellHaystack"><img src="https://cranlogs.r-pkg.org/badges/grand-total/singleCellHaystack" alt="CRAN Downloads"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
