<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application on toy example • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Application on toy example">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../articles/examples/a02_example_scRNAseq.html">Example on scRNA-seq</a>
    </li>
    <li>
      <a href="../articles/examples/a03_example_spatial_visium.html">Example on Spatial Visium</a>
    </li>
    <li>
      <a href="../articles/examples/a04_example_spatial_slideseqV2.html">Example on Spatial Slide-seqV2</a>
    </li>
    <li>
      <a href="../articles/examples/a05_moca_100k.html">Application to MOCA 100k cells</a>
    </li>
    <li>
      <a href="../articles/examples/a06_pseudotime.html">Application to pseudotime</a>
    </li>
    <li>
      <a href="../articles/examples/a07_gene_sets.html">Application to gene sets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Application on toy example</h1>
                        <h4 data-toc-skip class="author">Alexis
Vandenbon</h4>
            
            <h4 data-toc-skip class="date">2022-12-23</h4>
      
      
      <div class="hidden name"><code>a01_toy_example.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="application-on-a-toy-dataset">Application on a toy dataset<a class="anchor" aria-label="anchor" href="#application-on-a-toy-dataset"></a>
</h2>
<p>A small toy dataset is included in the package. The toy dataset
includes:</p>
<ul>
<li><p><code>dat.expression</code>: a toy scRNA-seq dataset with genes
(rows) and cells (columns)</p></li>
<li><p><code>dat.tsne</code>: 2D coordinates of the cells in a t-SNE
splot</p></li>
</ul>
<p>First, let’s apply <code>haystack</code> (the main function of the
package) on the toy dataset. This should take just several seconds on a
typical desktop computer.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">singleCellHaystack</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># run the main 'haystack' analysis</span></span>
<span><span class="co"># inputs are:</span></span>
<span><span class="co"># 1) the coordinates of the cells in the input space (here: dat.tsne)</span></span>
<span><span class="co"># 2) the expression data (dat.expression)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/haystack.html">haystack</a></span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="va">dat.expression</span><span class="op">)</span></span>
<span><span class="co">#&gt; ### calling haystack_continuous_highD()...</span></span>
<span><span class="co">#&gt; ### Using package sparseMatrixStats to speed up statistics in sparse matrices.</span></span>
<span><span class="co">#&gt; ### Calculating row-wise mean and SD...</span></span>
<span><span class="co">#&gt; ### Filtered 0 genes with zero variance...</span></span>
<span><span class="co">#&gt; ### Using 100 randomizations...</span></span>
<span><span class="co">#&gt; ### Using 100 genes to randomize...</span></span>
<span><span class="co">#&gt; Warning in haystack_continuous_highD(x, expression = expression,</span></span>
<span><span class="co">#&gt; weights.advanced.Q = weights.advanced.Q, : The value of 'grid.points' appears to</span></span>
<span><span class="co">#&gt; be very high (&gt; No. of cells / 10). You can set the number of grid points using</span></span>
<span><span class="co">#&gt; the 'grid.points' parameter.</span></span>
<span><span class="co">#&gt; ### scaling input data...</span></span>
<span><span class="co">#&gt; ### deciding grid points...</span></span>
<span><span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span></span>
<span><span class="co">#&gt; ### performing randomizations...</span></span>
<span><span class="co">#&gt; ### estimating p-values...</span></span>
<span><span class="co">#&gt; ### picking model for mean D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.09</span></span>
<span><span class="co">#&gt; ### best df    : 3</span></span>
<span><span class="co">#&gt; ### picking model for stdev D_KL...</span></span>
<span><span class="co">#&gt; ### using natural splines</span></span>
<span><span class="co">#&gt; ### best RMSD  : 0.02</span></span>
<span><span class="co">#&gt; ### best df    : 5</span></span>
<span><span class="co">#&gt; ### returning result...</span></span>
<span></span>
<span><span class="co"># the returned results 'res' is of class 'haystack'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "haystack"</span></span></code></pre></div>
<p>Let’s have a look at the most significant differentially expressed
genes (DEGs).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show top 10 DEGs</span></span>
<span><span class="fu"><a href="../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res</span>, n<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt;              D_KL log.p.vals log.p.adj</span></span>
<span><span class="co">#&gt; gene_79  2.447641  -39.94871 -37.24974</span></span>
<span><span class="co">#&gt; gene_497 2.271242  -39.84546 -37.14649</span></span>
<span><span class="co">#&gt; gene_242 1.742783  -35.91184 -33.21287</span></span>
<span><span class="co">#&gt; gene_275 1.819669  -35.88741 -33.18844</span></span>
<span><span class="co">#&gt; gene_62  2.174074  -35.63854 -32.93957</span></span>
<span><span class="co">#&gt; gene_71  2.546493  -34.83522 -32.13625</span></span>
<span><span class="co">#&gt; gene_381 2.733446  -34.38544 -31.68647</span></span>
<span><span class="co">#&gt; gene_351 1.844673  -33.45724 -30.75827</span></span>
<span><span class="co">#&gt; gene_479 2.343509  -31.98940 -29.29043</span></span>
<span><span class="co">#&gt; gene_300 2.097546  -30.42637 -27.72740</span></span>
<span></span>
<span><span class="co"># alternatively: use a p-value threshold</span></span>
<span><span class="co">#show_result_haystack(res.haystack = res, p.value.threshold = 1e-10)</span></span></code></pre></div>
<p>One of the most significant DEGs is “gene_497”. Here we visualize its
expression in the t-SNE plot. As you can see, this DEG is expressed only
in cells in the upper-left corner of the plot.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">d</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;            tSNE1     tSNE2 gene_1 gene_2</span></span>
<span><span class="co">#&gt; cell_1 -21.69304 11.599176      0      0</span></span>
<span><span class="co">#&gt; cell_2 -20.28140 10.808351      0      0</span></span>
<span><span class="co">#&gt; cell_3 -22.69715  8.643215      0      2</span></span>
<span><span class="co">#&gt; cell_4 -20.13836 12.485293      0      0</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">d</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">tSNE1</span>, <span class="va">tSNE2</span>, color<span class="op">=</span><span class="va">gene_497</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_distiller</a></span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Spectral"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a01_toy_example_files/figure-html/unnamed-chunk-2-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Yes, the coordinates of the cells in this toy example t-SNE space
roughly resemble a haystack; see <a href="https://en.wikipedia.org/wiki/Haystacks_(Monet_series)" class="external-link">the
Haystack paintings by Monet</a>.</p>
</div>
<div class="section level2">
<h2 id="clustering-and-visualization">Clustering and visualization<a class="anchor" aria-label="anchor" href="#clustering-and-visualization"></a>
</h2>
<p>You are not limited to single genes. Here, we pick up a set of DEGs,
and group them by their expression pattern in the plot into 5
clusters.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the top most significant genes, and cluster them by their distribution pattern in the 2D plot</span></span>
<span><span class="va">sorted.table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/show_result_haystack.html">show_result_haystack</a></span><span class="op">(</span>res.haystack <span class="op">=</span> <span class="va">res</span>, p.value.threshold <span class="op">=</span> <span class="fl">1e-10</span><span class="op">)</span></span>
<span><span class="va">gene.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">sorted.table</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># k-means clustering</span></span>
<span><span class="co">#km &lt;- kmeans_haystack(dat.tsne, dat.expression[gene.subset, ], grid.coordinates=res$info$grid.coordinates, k=5)</span></span>
<span><span class="co">#km.clusters &lt;- km$cluster</span></span>
<span></span>
<span><span class="co"># alternatively: hierarchical clustering</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hclust_haystack.html">hclust_haystack</a></span><span class="op">(</span><span class="va">dat.tsne</span>, <span class="va">dat.expression</span><span class="op">[</span><span class="va">gene.subset</span>, <span class="op">]</span>, grid.coordinates<span class="op">=</span><span class="va">res</span><span class="op">$</span><span class="va">info</span><span class="op">$</span><span class="va">grid.coordinates</span><span class="op">)</span></span>
<span><span class="co">#&gt; ### collecting density data...</span></span></code></pre></div>
<p>… and visualize the pattern of the selected genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">[</span><span class="va">gene.subset</span>, <span class="op">]</span>, show_column_names<span class="op">=</span><span class="cn">FALSE</span>, cluster_rows<span class="op">=</span><span class="va">hm</span>, name<span class="op">=</span><span class="st">"expression"</span><span class="op">)</span></span></code></pre></div>
<p><img src="a01_toy_example_files/figure-html/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;"></p>
<p>We divide the genes into clusters with cutree.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hm.clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html" class="external-link">cutree</a></span><span class="op">(</span><span class="va">hm</span>, k<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">hm.clusters</span><span class="op">)</span></span>
<span><span class="co">#&gt; hm.clusters</span></span>
<span><span class="co">#&gt;  1  2  3  4 </span></span>
<span><span class="co">#&gt;  8 16  7 17</span></span></code></pre></div>
<p>Then calculate the average expression of the genes in each
cluster.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">cluster</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">hm.clusters</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">d</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cluster_"</span>, <span class="va">cluster</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">dat.expression</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">hm.clusters</span> <span class="op">==</span> <span class="va">cluster</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster_1"</span>, <span class="st">"cluster_2"</span>, <span class="st">"cluster_3"</span>, <span class="st">"cluster_4"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">d</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">tSNE1</span>, <span class="va">tSNE2</span>, color<span class="op">=</span><span class="va">.data</span><span class="op">[[</span><span class="va">cluster</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_distiller</a></span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Spectral"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a01_toy_example_files/figure-html/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;"></p>
<p>From this plot we can see that genes in each cluster are expressed in
different subsets of cells.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
