<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application on toy example • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Application on toy example">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a01_toy_example.html">Toy example</a>
    </li>
    <li>
      <a href="../articles/examples/a02_example_highD_default.html">Multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../articles/examples/a03_example_highD_advanced.html">Advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../articles/examples/a04_example_spatial_transcriptomics.html">Spatial transcriptomics</a>
    </li>
    <li>
      <a href="../articles/examples/a05_moca_100k.html">MOCA 100k cells</a>
    </li>
    <li>
      <a href="../articles/examples/a06_example_tsne2D_default.html">2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../articles/examples/a07_example_tsne2D_advanced.html">Advanced mode on 2D t-SNE coordinates</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a01_toy_example_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Application on toy example</h1>
                        <h4 class="author">Alexis Vandenbon</h4>
            
            <h4 class="date">2020-09-17</h4>
      
      
      <div class="hidden name"><code>a01_toy_example.Rmd</code></div>

    </div>

    
    
<div id="application-on-a-toy-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#application-on-a-toy-dataset" class="anchor"></a>Application on a toy dataset</h1>
<p>A small toy dataset is included in the package. The toy dataset includes:</p>
<ul>
<li><p><code>dat.expression</code>: a toy scRNA-seq dataset with genes (rows) and cells (columns)</p></li>
<li><p><code>dat.tsne</code>: 2D coordinates of the cells in a t-SNE splot</p></li>
</ul>
<p>First, let’s apply <code>haystack</code> (the main function of the package) on the toy dataset. This should take just several seconds on a typical desktop computer.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">singleCellHaystack</span>)
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">1234</span>)

<span class="co"># Turn the expression data into detection (gene detected = TRUE, not detected = FALSE)</span>
<span class="co"># let's define detection as having more than 1 read</span>
<span class="kw">dat.detection</span> <span class="op">&lt;-</span> <span class="kw">dat.expression</span> <span class="op">&gt;</span> <span class="fl">1</span>

<span class="co"># run the main 'haystack' analysis</span>
<span class="co"># inputs are:</span>
<span class="co"># 1) the coordinates of the cells in the input space (here: dat.tsne)</span>
<span class="co"># 2) the detection data (dat.detection)</span>
<span class="co"># 3) the method used ("2D" since we are using a 2D input space here)</span>
<span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/haystack.html">haystack</a></span>(<span class="kw">dat.tsne</span>, detection = <span class="kw">dat.detection</span>, method = <span class="st">"2D"</span>)
<span class="co">#&gt; ### calling haystack_2D()...</span>
<span class="co">#&gt; ### setting parameters...</span>
<span class="co">#&gt; ### calculating Kullback-Leibler divergences...</span>
<span class="co">#&gt; ### performing randomizations...</span>
<span class="co">#&gt; ### estimating p-values...</span>
<span class="co">#&gt; ### returning result...</span>

<span class="co"># the returned results 'res' is of class 'haystack'</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="kw">res</span>)
<span class="co">#&gt; [1] "haystack"</span>
</pre></div>
<p>Let’s have a look at the most significant differentially expressed genes (DEGs).</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co"># show top 10 DEGs</span>
<span class="fu"><a href="../reference/show_result_haystack.html">show_result_haystack</a></span>(res.haystack = <span class="kw">res</span>, n=<span class="fl">10</span>)
<span class="co">#&gt;              D_KL log.p.vals log.p.adj T.counts</span>
<span class="co">#&gt; gene_497 1.599646  -25.55327 -22.85430       86</span>
<span class="co">#&gt; gene_79  1.929800  -24.93934 -22.24037       68</span>
<span class="co">#&gt; gene_62  1.519993  -24.68869 -21.98972       81</span>
<span class="co">#&gt; gene_61  1.470475  -24.12825 -21.42928       80</span>
<span class="co">#&gt; gene_339 1.414814  -23.87108 -21.17211       87</span>
<span class="co">#&gt; gene_351 1.408946  -23.62384 -20.92487       89</span>
<span class="co">#&gt; gene_137 1.349994  -23.08019 -20.38122       89</span>
<span class="co">#&gt; gene_444 1.435882  -22.92343 -20.22446       75</span>
<span class="co">#&gt; gene_317 1.386194  -22.38427 -19.68530       94</span>
<span class="co">#&gt; gene_313 1.294544  -22.16986 -19.47089       78</span>

<span class="co"># alternatively: use a p-value threshold</span>
<span class="co">#show_result_haystack(res.haystack = res, p.value.threshold = 1e-10)</span>
</pre></div>
<p>One of the most significant DEGs is “gene_497”. Here we visualize its expression in the t-SNE plot. As you can see, this DEG is expressed only in cells in the upper-left corner of the plot.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co"># visualize one of the surprizing genes</span>
<span class="fu"><a href="../reference/plot_gene_haystack.html">plot_gene_haystack</a></span>(
  <span class="kw">dat.tsne</span>,
  expression = <span class="kw">dat.expression</span>,
  gene = <span class="st">"gene_497"</span>,
  detection = <span class="kw">dat.detection</span>,
  high.resolution = <span class="fl">TRUE</span>,
  point.size = <span class="fl">2</span>
)
</pre></div>
<p><img src="a01_toy_example_files/figure-html/example3-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Yes, the coordinates of the cells in this toy example t-SNE space roughly resemble a haystack; see <a href="https://en.wikipedia.org/wiki/Haystacks_(Monet_series)">the Haystack paintings by Monet</a>.</p>
</div>
<div id="clustering-and-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering-and-visualization" class="anchor"></a>Clustering and visualization</h1>
<p>You are not limited to single genes. Here, we pick up a set of DEGs, and group them by their expression pattern in the plot into 5 clusters.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># get the top most significant genes, and cluster them by their distribution pattern in the 2D plot</span>
<span class="kw">sorted.table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/show_result_haystack.html">show_result_haystack</a></span>(res.haystack = <span class="kw">res</span>, p.value.threshold = <span class="fl">1e-10</span>)
<span class="kw">gene.subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(<span class="kw">sorted.table</span>)

<span class="co"># k-means clustering</span>
<span class="kw">km</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/kmeans_haystack.html">kmeans_haystack</a></span>(<span class="kw">dat.tsne</span>, detection=<span class="kw">dat.detection</span>, genes=<span class="kw">gene.subset</span>, k=<span class="fl">5</span>)
<span class="kw">km.clusters</span> <span class="op">&lt;-</span> <span class="kw">km</span><span class="op">$</span><span class="kw">cluster</span>

<span class="co"># alternatively: hierarchical clustering</span>
<span class="co">#hc &lt;- hclust_haystack(dat.tsne, detection=dat.detection, genes=gene.subset)</span>
<span class="co">#hc.clusters &lt;- cutree(hc,k = 5)</span>
</pre></div>
<p>… and visualize the average pattern of the genes in cluster 1 (for example).</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># visualize cluster distributions</span>
<span class="fu"><a href="../reference/plot_gene_set_haystack.html">plot_gene_set_haystack</a></span>(<span class="kw">dat.tsne</span>, detection = <span class="kw">dat.detection</span>, 
                        genes=<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">km.clusters</span>[<span class="kw">km.clusters</span><span class="op">==</span><span class="fl">1</span>]), point.size=<span class="fl">2</span>)
</pre></div>
<p><img src="a01_toy_example_files/figure-html/example5-1.png" width="576" style="display: block; margin: auto;"></p>
<p>From this plot we can see that genes in cluster 1 are mainly expressed in cells in the lower right part of the plot.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
