<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Application of 'advanced' mode on multi-dimensional coordinates • singleCellHaystack</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Application of 'advanced' mode on multi-dimensional coordinates">
<meta property="og:description" content="singleCellHaystack">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellHaystack</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a01_toy_example.html">Application on toy example</a>
    </li>
    <li>
      <a href="../articles/a02_example_highD_default.html">Application on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../articles/a03_example_highD_advanced.html">Application of the advanced mode on multi-dimensional coordinates</a>
    </li>
    <li>
      <a href="../articles/a04_example_tsne2D_default.html">Application on 2D t-SNE coordinates</a>
    </li>
    <li>
      <a href="../articles/a05_example_tsne2D_advanced.html">Application of the advanced mode on 2D t-SNE coordinates</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alexisvdb/singleCellHaystack">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Application of ‘advanced’ mode on multi-dimensional coordinates</h1>
                        <h4 class="author">Alexis Vandenbon</h4>
            
            <h4 class="date">2020-04-15</h4>
      
      
      <div class="hidden name"><code>a03_example_highD_advanced.Rmd</code></div>

    </div>

    
    
<div id="tldr" class="section level1">
<h1 class="hasAnchor">
<a href="#tldr" class="anchor"></a>tl;dr</h1>
<p>You can indicate the general expression levels of cells using input parameter <code>use.advanced.sampling</code>. If you do that,<code>singleCellHaystack</code> takes the general expression levels into account and finds genes that are different from that general expression pattern.</p>
</div>
<div id="preparing-the-input-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preparing-the-input-data" class="anchor"></a>Preparing the input data</h1>
<p>The data used in these examples can be found <a href="https://genomics.virus.kyoto-u.ac.jp/alexisvdb/singleCellHaystack/">here</a>. We recommend downloading the .rda file and loading it using the <code><a href="https://rdrr.io/r/base/load.html">load()</a></code> function. You can also download the individual data files separately.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r">load(url("https://genomics.virus.kyoto-u.ac.jp/alexisvdb/singleCellHaystack/tabula_muris_marrow_P7_2.rda"))
ls()
#&gt; [1] "dat.detection"  "dat.expression" "dat.pca"        "dat.tsne"      
#&gt; [5] "dat.umap"</pre></body></html></div>
<p>This data should include the following objects:</p>
<ul>
<li>
<code>dat.expression</code>: a matrix object with the expression of genes (rows) in each cell (columns).</li>
<li>
<code>dat.detection</code>: a logical matrix object with the detection of genes (rows) in each cell (columns). TRUE means detected, FALSE means not detected.</li>
<li>
<code>dat.pca</code>: the output of PCA. This data.frame contains thefirst 50 pricipal components.</li>
<li>
<code>dat.tsne</code>: a data.frame with t-SNE coordinates (2D).</li>
<li>
<code>dat.umap</code>: a data.frame with UMAP coordinates (2D).</li>
</ul>
<p>Let’s have a look at the t-SNE plot, colouring cells be the number of genes detected in each cell:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r">general.detection </pre></body></html></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example1-1.png" width="576" style="display: block; margin: auto;"> There are several groups of cells,although the borders between are not clear and several might consist of additional subclusters.</p>
<p>Moreover, the cells in the bottom right have in general fewer detected genes. If a gene is detected predominantly in these cells and not in others, this tendency goes against the general trend in expression. Such a trend would be especially surprizing.</p>
<p>We can use <code>singleCellHaystack</code> to find genes with biased expression patterns, taking into account general gene expression trends, using the option <code>use.advanced.sampling</code>.</p>
</div>
<div id="running-haystack-with-use-advanced-sampling" class="section level1">
<h1 class="hasAnchor">
<a href="#running-haystack-with-use-advanced-sampling" class="anchor"></a>Running <code>haystack</code> with <code>use.advanced.sampling</code>
</h1>
<p>First, load the package, and set a random seed to ensure replicability.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r">library(singleCellHaystack)
#&gt; 
#&gt; Attaching package: 'singleCellHaystack'
#&gt; The following objects are masked _by_ '.GlobalEnv':
#&gt; 
#&gt;     dat.expression, dat.tsne
set.seed(1)</pre></body></html></div>
<p>Next, run <code>haystack</code> on the first 10 principal components. Since the space is 10-dimensional, we set ‘method’ to ‘highD’. We also give the detection values as input to ‘detection’. This example dataset is relatively small, containing 1,981 cells, so running ‘haystack’ should take just 1 to 3 minutes to finish. We also give the detection values as input to ‘detection’. <code>use.advanced.sampling</code> is set to the vector of detected genes per cell, <code>general.detection</code>. By doing this, <code>haystack</code> will calculate cell distributions and perform randomizations taking into account the general detection levels.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r">res.pc10.adv  ### calling haystack_highD()...
#&gt; ### scaling input data...
#&gt; ### deciding grid points...
#&gt; ### calculating Kullback-Leibler divergences...
#&gt; ### ... 1000 features out of 12030 done
#&gt; ### ... 2000 features out of 12030 done
#&gt; ### ... 3000 features out of 12030 done
#&gt; ### ... 4000 features out of 12030 done
#&gt; ### ... 5000 features out of 12030 done
#&gt; ### ... 6000 features out of 12030 done
#&gt; ### ... 7000 features out of 12030 done
#&gt; ### ... 8000 features out of 12030 done
#&gt; ### ... 9000 features out of 12030 done
#&gt; ### ... 10000 features out of 12030 done
#&gt; ### ... 11000 features out of 12030 done
#&gt; ### ... 12000 features out of 12030 done
#&gt; ### starting randomizations...
#&gt; ### ... 10 values out of 200 done
#&gt; ### ... 20 values out of 200 done
#&gt; ### ... 30 values out of 200 done
#&gt; ### ... 40 values out of 200 done
#&gt; ### ... 50 values out of 200 done
#&gt; ### ... 60 values out of 200 done
#&gt; ### ... 70 values out of 200 done
#&gt; ### ... 80 values out of 200 done
#&gt; ### ... 90 values out of 200 done
#&gt; ### ... 100 values out of 200 done
#&gt; ### ... 110 values out of 200 done
#&gt; ### ... 120 values out of 200 done
#&gt; ### ... 130 values out of 200 done
#&gt; ### ... 140 values out of 200 done
#&gt; ### ... 150 values out of 200 done
#&gt; ### ... 160 values out of 200 done
#&gt; ### ... 170 values out of 200 done
#&gt; ### ... 180 values out of 200 done
#&gt; ### ... 190 values out of 200 done
#&gt; ### ... 200 values out of 200 done
#&gt; ### estimating p-values...
#&gt; ### returning result...</pre></body></html></div>
<p>Let’s have a look at the most biased genes. The most strongly biased gene is Snca. We can plot the expression an detection of this gene using the <code>plot_gene_haystack</code> function.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r">show_result_haystack(res.haystack = res.pc10.adv, n = 5)
#&gt;                   D_KL log.p.vals log.p.adj T.counts
#&gt; Snca          2.303152  -1425.515 -1421.435      474
#&gt; Gypa          2.248503  -1341.622 -1337.541      448
#&gt; Hba-a1        2.176333  -1321.308 -1317.228      479
#&gt; Trim10        2.396674  -1157.922 -1153.841      368
#&gt; 5730469M10Rik 2.048861  -1151.420 -1147.340      435
# plotting detection of this gene
plot_gene_haystack(x = dat.tsne, gene = "Snca", expression = dat.detection)</pre></body></html></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example4-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="co"># plotting log expression of this gene</span>
<span class="fu"><a href="../reference/plot_gene_haystack.html">plot_gene_haystack</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">dat.tsne</span>, <span class="kw">gene</span> <span class="kw">=</span> <span class="st">"Snca"</span>, <span class="kw">expression</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">dat.expression</span>))</pre></body></html></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example4-2.png" width="576" style="display: block; margin: auto;"></p>
<p>From the plots we can see that Snca is detected in the lower right groups of cells. As seen above, these cells have in general fewer detected genes, so the pattern of Snca is especially surprizing: it is deteced in cells that express in general few genes.</p>
<p>This explains why the advanced mode of <code>singleCellHaystack</code> judges Snca to be the most significantly biased.</p>
<p>Compare the pattern of Snca with that of the most biased gene in the default mode (see <a href="articles/examplehighD_default.html">here</a>). The most biased gene found in the “default” mode is strongly biased, but is detected in cells that have in general many detected genes.</p>
</div>
<div id="clustering-and-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering-and-visualization" class="anchor"></a>Clustering and visualization</h1>
<p>Next, let’s take the top 1000 biased genes, and cluster them by their expression pattern in the input space (first 10 PCs). Here we use <code>hclust_haystack</code>, which uses hierarchical clustering. Alternatively, we could use <code>kmeans_haystack</code> for k-means clustering.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"># get the top 1000 biased genes in the result
res.top </pre></body></html></div>
<p><code>hclust_haystack</code> returns as result a <code>hclust</code> tree, which we can cut into clusters using the <code>cutree</code> function. Here, we arbitrarily set the number of clusters to 5.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r">res.hc.clusters  res.hc.clusters
#&gt;   1   2   3   4   5 
#&gt; 145 473 196 103  83</pre></body></html></div>
<p>Let’s run through the 5 clusters and plot their averaged detection pattern using <code>plot_gene_set_haystack</code>, which is similar to <code>plot_gene_haystack</code> but uses a set of genes as input instead of just 1 gene.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r">pl </pre></body></html></div>
<p><img src="a03_example_highD_advanced_files/figure-html/example7-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">res.hc.clusters</span>[<span class="st">"Snca"</span>] <span class="co"># the most biased gene is in cluster 5</span>
<span class="co">#&gt; Snca </span>
<span class="co">#&gt;    5</span></pre></body></html></div>
<p>The most biased genes, Snca, was clustered into cluster 5. Comparing its expression pattern (see above) with that of each cluster, we can indeed see that it fits most closely with that of cluster 5. Cluster 5 genes are on average detected almost exclusively in the bottom right group of cells.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexis Vandenbon, Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
